error[E0502]: cannot borrow `x` as immutable because it is also borrowed as mutable
  --> $DIR/borrowck-report-with-custom-diagnostic.rs:17:14
   |
15 |     let y = &mut x;
   |                  - mutable borrow occurs here
16 |     //~^ mutable borrow occurs here
17 |     let z = &x; //~ ERROR cannot borrow
   |              ^ immutable borrow occurs here
18 |     //~^ immutable borrow occurs here
19 | }
   | - mutable borrow ends here

error[E0502]: cannot borrow `x` as mutable because it is also borrowed as immutable
  --> $DIR/borrowck-report-with-custom-diagnostic.rs:29:26
   |
27 |             let y = &x;
   |                      - immutable borrow occurs here
28 |             //~^ immutable borrow occurs here
29 |             let z = &mut x; //~ ERROR cannot borrow
   |                          ^ mutable borrow occurs here
30 |             //~^ mutable borrow occurs here
31 |         }
   |         - immutable borrow ends here

error[E0499]: cannot borrow `x` as mutable more than once at a time
  --> $DIR/borrowck-report-with-custom-diagnostic.rs:43:22
   |
41 |         let y = &mut x;
   |                      - first mutable borrow occurs here
42 |         //~^ first mutable borrow occurs here
43 |         let z = &mut x; //~ ERROR cannot borrow
   |                      ^ second mutable borrow occurs here
44 |         //~^ second mutable borrow occurs here
45 |     };
   |     - first borrow ends here

error: aborting due to 3 previous errors

