error: duplicate `if_let` feature attribute
  --> $DIR/duplicate-features.rs:18:12
   |
LL | #![feature(if_let)] //~ ERROR duplicate `if_let` feature attribute
   |            ^^^^^^
   |
note: lint level defined here
  --> $DIR/duplicate-features.rs:12:9
   |
LL | #![deny(duplicate_features)]
   |         ^^^^^^^^^^^^^^^^^^

error: duplicate `rust1` feature attribute
  --> $DIR/duplicate-features.rs:15:12
   |
LL | #![feature(rust1)] //~ ERROR duplicate `rust1` feature attribute
   |            ^^^^^

error: aborting due to 2 previous errors

