error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:50:17
   |
LL |     let X(_t) = *s;
   |           --    ^^
   |           |     |
   |           |     cannot move out of borrowed content
   |           |     help: consider removing the `*`: `s`
   |           data moved here
   |
note: move occurs because `_t` has type `Y`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:50:11
   |
LL |     let X(_t) = *s;
   |           ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:54:30
   |
LL |     if let Either::One(_t) = *r { }
   |                        --    ^^
   |                        |     |
   |                        |     cannot move out of borrowed content
   |                        |     help: consider removing the `*`: `r`
   |                        data moved here
   |
note: move occurs because `_t` has type `X`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:54:24
   |
LL |     if let Either::One(_t) = *r { }
   |                        ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:58:33
   |
LL |     while let Either::One(_t) = *r { }
   |                           --    ^^
   |                           |     |
   |                           |     cannot move out of borrowed content
   |                           |     help: consider removing the `*`: `r`
   |                           data moved here
   |
note: move occurs because `_t` has type `X`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:58:27
   |
LL |     while let Either::One(_t) = *r { }
   |                           ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:62:11
   |
LL |     match *r {
   |           ^^
   |           |
   |           cannot move out of borrowed content
   |           help: consider removing the `*`: `r`
...
LL |         Either::One(_t)
   |                     -- data moved here
   |
note: move occurs because `_t` has type `X`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:66:21
   |
LL |         Either::One(_t)
   |                     ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:69:11
   |
LL |     match *r {
   |           ^^
   |           |
   |           cannot move out of borrowed content
   |           help: consider removing the `*`: `r`
...
LL |         Either::One(_t) => (),
   |                     -- data moved here
   |
note: move occurs because `_t` has type `X`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:73:21
   |
LL |         Either::One(_t) => (),
   |                     ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:78:17
   |
LL |     let X(_t) = *sm;
   |           --    ^^^
   |           |     |
   |           |     cannot move out of borrowed content
   |           |     help: consider removing the `*`: `sm`
   |           data moved here
   |
note: move occurs because `_t` has type `Y`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:78:11
   |
LL |     let X(_t) = *sm;
   |           ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:82:30
   |
LL |     if let Either::One(_t) = *rm { }
   |                        --    ^^^
   |                        |     |
   |                        |     cannot move out of borrowed content
   |                        |     help: consider removing the `*`: `rm`
   |                        data moved here
   |
note: move occurs because `_t` has type `X`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:82:24
   |
LL |     if let Either::One(_t) = *rm { }
   |                        ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:86:33
   |
LL |     while let Either::One(_t) = *rm { }
   |                           --    ^^^
   |                           |     |
   |                           |     cannot move out of borrowed content
   |                           |     help: consider removing the `*`: `rm`
   |                           data moved here
   |
note: move occurs because `_t` has type `X`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:86:27
   |
LL |     while let Either::One(_t) = *rm { }
   |                           ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:90:11
   |
LL |     match *rm {
   |           ^^^
   |           |
   |           cannot move out of borrowed content
   |           help: consider removing the `*`: `rm`
...
LL |         Either::One(_t)
   |                     -- data moved here
   |
note: move occurs because `_t` has type `X`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:94:21
   |
LL |         Either::One(_t)
   |                     ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:97:11
   |
LL |     match *rm {
   |           ^^^
   |           |
   |           cannot move out of borrowed content
   |           help: consider removing the `*`: `rm`
...
LL |         Either::One(_t) => (),
   |                     -- data moved here
   |
note: move occurs because `_t` has type `X`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:101:21
   |
LL |         Either::One(_t) => (),
   |                     ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:105:11
   |
LL |     match *rm {
   |           ^^^
   |           |
   |           cannot move out of borrowed content
   |           help: consider removing the `*`: `rm`
...
LL |         Either::One(_t) => (),
   |                     -- data moved here
   |
note: move occurs because `_t` has type `X`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:109:21
   |
LL |         Either::One(_t) => (),
   |                     ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:114:17
   |
LL |     let X(_t) = vs[0];
   |           --    ^^^^^
   |           |     |
   |           |     cannot move out of borrowed content
   |           |     help: consider borrowing here: `&vs[0]`
   |           data moved here
   |
note: move occurs because `_t` has type `Y`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:114:11
   |
LL |     let X(_t) = vs[0];
   |           ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:118:30
   |
LL |     if let Either::One(_t) = vr[0] { }
   |                        --    ^^^^^
   |                        |     |
   |                        |     cannot move out of borrowed content
   |                        |     help: consider borrowing here: `&vr[0]`
   |                        data moved here
   |
note: move occurs because `_t` has type `X`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:118:24
   |
LL |     if let Either::One(_t) = vr[0] { }
   |                        ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:122:33
   |
LL |     while let Either::One(_t) = vr[0] { }
   |                           --    ^^^^^
   |                           |     |
   |                           |     cannot move out of borrowed content
   |                           |     help: consider borrowing here: `&vr[0]`
   |                           data moved here
   |
note: move occurs because `_t` has type `X`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:122:27
   |
LL |     while let Either::One(_t) = vr[0] { }
   |                           ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:126:11
   |
LL |     match vr[0] {
   |           ^^^^^
   |           |
   |           cannot move out of borrowed content
   |           help: consider borrowing here: `&vr[0]`
...
LL |         Either::One(_t)
   |                     -- data moved here
   |
note: move occurs because `_t` has type `X`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:130:21
   |
LL |         Either::One(_t)
   |                     ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:133:11
   |
LL |     match vr[0] {
   |           ^^^^^
   |           |
   |           cannot move out of borrowed content
   |           help: consider borrowing here: `&vr[0]`
...
LL |         Either::One(_t) => (),
   |                     -- data moved here
   |
note: move occurs because `_t` has type `X`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:137:21
   |
LL |         Either::One(_t) => (),
   |                     ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:142:17
   |
LL |     let X(_t) = vsm[0];
   |           --    ^^^^^^
   |           |     |
   |           |     cannot move out of borrowed content
   |           |     help: consider borrowing here: `&vsm[0]`
   |           data moved here
   |
note: move occurs because `_t` has type `Y`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:142:11
   |
LL |     let X(_t) = vsm[0];
   |           ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:146:30
   |
LL |     if let Either::One(_t) = vrm[0] { }
   |                        --    ^^^^^^
   |                        |     |
   |                        |     cannot move out of borrowed content
   |                        |     help: consider borrowing here: `&vrm[0]`
   |                        data moved here
   |
note: move occurs because `_t` has type `X`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:146:24
   |
LL |     if let Either::One(_t) = vrm[0] { }
   |                        ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:150:33
   |
LL |     while let Either::One(_t) = vrm[0] { }
   |                           --    ^^^^^^
   |                           |     |
   |                           |     cannot move out of borrowed content
   |                           |     help: consider borrowing here: `&vrm[0]`
   |                           data moved here
   |
note: move occurs because `_t` has type `X`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:150:27
   |
LL |     while let Either::One(_t) = vrm[0] { }
   |                           ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:154:11
   |
LL |     match vrm[0] {
   |           ^^^^^^
   |           |
   |           cannot move out of borrowed content
   |           help: consider borrowing here: `&vrm[0]`
...
LL |         Either::One(_t)
   |                     -- data moved here
   |
note: move occurs because `_t` has type `X`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:158:21
   |
LL |         Either::One(_t)
   |                     ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:161:11
   |
LL |     match vrm[0] {
   |           ^^^^^^
   |           |
   |           cannot move out of borrowed content
   |           help: consider borrowing here: `&vrm[0]`
...
LL |         Either::One(_t) => (),
   |                     -- data moved here
   |
note: move occurs because `_t` has type `X`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:165:21
   |
LL |         Either::One(_t) => (),
   |                     ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:169:11
   |
LL |     match vrm[0] {
   |           ^^^^^^
   |           |
   |           cannot move out of borrowed content
   |           help: consider borrowing here: `&vrm[0]`
...
LL |         Either::One(_t) => (),
   |                     -- data moved here
   |
note: move occurs because `_t` has type `X`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:173:21
   |
LL |         Either::One(_t) => (),
   |                     ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:180:18
   |
LL |     let &X(_t) = s;
   |         ------   ^ cannot move out of borrowed content
   |         |  |
   |         |  data moved here
   |         help: consider removing the `&`: `X(_t)`
   |
note: move occurs because `_t` has type `Y`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:180:12
   |
LL |     let &X(_t) = s;
   |            ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:184:31
   |
LL |     if let &Either::One(_t) = r { }
   |            ----------------   ^ cannot move out of borrowed content
   |            |            |
   |            |            data moved here
   |            help: consider removing the `&`: `Either::One(_t)`
   |
note: move occurs because `_t` has type `X`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:184:25
   |
LL |     if let &Either::One(_t) = r { }
   |                         ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:188:34
   |
LL |     while let &Either::One(_t) = r { }
   |               ----------------   ^ cannot move out of borrowed content
   |               |            |
   |               |            data moved here
   |               help: consider removing the `&`: `Either::One(_t)`
   |
note: move occurs because `_t` has type `X`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:188:28
   |
LL |     while let &Either::One(_t) = r { }
   |                            ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:192:11
   |
LL |     match r {
   |           ^ cannot move out of borrowed content
LL |         //~^ ERROR cannot move
LL |         &Either::One(_t)
   |         ----------------
   |         |            |
   |         |            data moved here
   |         help: consider removing the `&`: `Either::One(_t)`
   |
note: move occurs because `_t` has type `X`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:194:22
   |
LL |         &Either::One(_t)
   |                      ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:200:11
   |
LL |     match r {
   |           ^ cannot move out of borrowed content
LL |         //~^ ERROR cannot move
LL |         &Either::One(_t) => (),
   |         ----------------
   |         |            |
   |         |            data moved here
   |         help: consider removing the `&`: `Either::One(_t)`
   |
note: move occurs because `_t` has type `X`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:202:22
   |
LL |         &Either::One(_t) => (),
   |                      ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:207:11
   |
LL |     match r {
   |           ^ cannot move out of borrowed content
LL |         //~^ ERROR cannot move
LL |         &Either::One(_t) => (),
   |         ----------------
   |         |            |
   |         |            data moved here
   |         help: consider removing the `&`: `Either::One(_t)`
   |
note: move occurs because `_t` has type `X`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:209:22
   |
LL |         &Either::One(_t) => (),
   |                      ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:219:22
   |
LL |     let &mut X(_t) = sm;
   |         ----------   ^^ cannot move out of borrowed content
   |         |      |
   |         |      data moved here
   |         help: consider removing the `&mut`: `X(_t)`
   |
note: move occurs because `_t` has type `Y`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:219:16
   |
LL |     let &mut X(_t) = sm;
   |                ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:223:35
   |
LL |     if let &mut Either::One(_t) = rm { }
   |            --------------------   ^^ cannot move out of borrowed content
   |            |                |
   |            |                data moved here
   |            help: consider removing the `&mut`: `Either::One(_t)`
   |
note: move occurs because `_t` has type `X`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:223:29
   |
LL |     if let &mut Either::One(_t) = rm { }
   |                             ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:227:38
   |
LL |     while let &mut Either::One(_t) = rm { }
   |               --------------------   ^^ cannot move out of borrowed content
   |               |                |
   |               |                data moved here
   |               help: consider removing the `&mut`: `Either::One(_t)`
   |
note: move occurs because `_t` has type `X`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:227:32
   |
LL |     while let &mut Either::One(_t) = rm { }
   |                                ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:231:11
   |
LL |     match rm {
   |           ^^ cannot move out of borrowed content
LL |         //~^ ERROR cannot move
LL |         &mut Either::One(_t) => (),
   |                          -- data moved here
...
LL |         &mut Either::Two(_t) => (),
   |                          -- ...and here
   |
note: move occurs because these variables have types that don't implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:233:26
   |
LL |         &mut Either::One(_t) => (),
   |                          ^^
...
LL |         &mut Either::Two(_t) => (),
   |                          ^^
help: consider removing the `&mut`
   |
LL |         Either::One(_t) => (),
   |         ^^^^^^^^^^^^^^^
help: consider removing the `&mut`
   |
LL |         Either::Two(_t) => (),
   |         ^^^^^^^^^^^^^^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:240:11
   |
LL |     match rm {
   |           ^^ cannot move out of borrowed content
LL |         //~^ ERROR cannot move
LL |         &mut Either::One(_t) => (),
   |         --------------------
   |         |                |
   |         |                data moved here
   |         help: consider removing the `&mut`: `Either::One(_t)`
   |
note: move occurs because `_t` has type `X`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:242:26
   |
LL |         &mut Either::One(_t) => (),
   |                          ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:247:11
   |
LL |     match rm {
   |           ^^ cannot move out of borrowed content
LL |         //~^ ERROR cannot move
LL |         &mut Either::One(_t) => (),
   |         --------------------
   |         |                |
   |         |                data moved here
   |         help: consider removing the `&mut`: `Either::One(_t)`
   |
note: move occurs because `_t` has type `X`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:249:26
   |
LL |         &mut Either::One(_t) => (),
   |                          ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:254:11
   |
LL |     match rm {
   |           ^^ cannot move out of borrowed content
LL |         //~^ ERROR cannot move
LL |         &mut Either::One(_t) => (),
   |         --------------------
   |         |                |
   |         |                data moved here
   |         help: consider removing the `&mut`: `Either::One(_t)`
   |
note: move occurs because `_t` has type `X`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:256:26
   |
LL |         &mut Either::One(_t) => (),
   |                          ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:270:21
   |
LL |     let (&X(_t),) = (&x.clone(),);
   |             --      ^^^^^^^^^^^^^ cannot move out of borrowed content
   |             |
   |             data moved here
   |
note: move occurs because `_t` has type `Y`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:270:13
   |
LL |     let (&X(_t),) = (&x.clone(),);
   |             ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:272:34
   |
LL |     if let (&Either::One(_t),) = (&e.clone(),) { }
   |                          --      ^^^^^^^^^^^^^ cannot move out of borrowed content
   |                          |
   |                          data moved here
   |
note: move occurs because `_t` has type `X`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:272:26
   |
LL |     if let (&Either::One(_t),) = (&e.clone(),) { }
   |                          ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:274:37
   |
LL |     while let (&Either::One(_t),) = (&e.clone(),) { }
   |                             --      ^^^^^^^^^^^^^ cannot move out of borrowed content
   |                             |
   |                             data moved here
   |
note: move occurs because `_t` has type `X`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:274:29
   |
LL |     while let (&Either::One(_t),) = (&e.clone(),) { }
   |                             ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:276:11
   |
LL |     match (&e.clone(),) {
   |           ^^^^^^^^^^^^^ cannot move out of borrowed content
LL |         //~^ ERROR cannot move
LL |         (&Either::One(_t),)
   |                       -- data moved here
   |
note: move occurs because `_t` has type `X`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:278:23
   |
LL |         (&Either::One(_t),)
   |                       ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:284:25
   |
LL |     let (&mut X(_t),) = (&mut xm.clone(),);
   |                 --      ^^^^^^^^^^^^^^^^^^ cannot move out of borrowed content
   |                 |
   |                 data moved here
   |
note: move occurs because `_t` has type `Y`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:284:17
   |
LL |     let (&mut X(_t),) = (&mut xm.clone(),);
   |                 ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:286:38
   |
LL |     if let (&mut Either::One(_t),) = (&mut em.clone(),) { }
   |                              --      ^^^^^^^^^^^^^^^^^^ cannot move out of borrowed content
   |                              |
   |                              data moved here
   |
note: move occurs because `_t` has type `X`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:286:30
   |
LL |     if let (&mut Either::One(_t),) = (&mut em.clone(),) { }
   |                              ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:288:41
   |
LL |     while let (&mut Either::One(_t),) = (&mut em.clone(),) { }
   |                                 --      ^^^^^^^^^^^^^^^^^^ cannot move out of borrowed content
   |                                 |
   |                                 data moved here
   |
note: move occurs because `_t` has type `X`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:288:33
   |
LL |     while let (&mut Either::One(_t),) = (&mut em.clone(),) { }
   |                                 ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:290:11
   |
LL |     match (&mut em.clone(),) {
   |           ^^^^^^^^^^^^^^^^^^ cannot move out of borrowed content
LL |         //~^ ERROR cannot move
LL |         (&mut Either::One(_t),) => (),
   |                           -- data moved here
LL |         (&mut Either::Two(_t),) => (),
   |                           -- ...and here
   |
note: move occurs because these variables have types that don't implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:292:27
   |
LL |         (&mut Either::One(_t),) => (),
   |                           ^^
LL |         (&mut Either::Two(_t),) => (),
   |                           ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:300:18
   |
LL |     let &X(_t) = &x;
   |         ------   ^^ cannot move out of borrowed content
   |         |  |
   |         |  data moved here
   |         help: consider removing the `&`: `X(_t)`
   |
note: move occurs because `_t` has type `Y`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:300:12
   |
LL |     let &X(_t) = &x;
   |            ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:304:31
   |
LL |     if let &Either::One(_t) = &e { }
   |            ----------------   ^^ cannot move out of borrowed content
   |            |            |
   |            |            data moved here
   |            help: consider removing the `&`: `Either::One(_t)`
   |
note: move occurs because `_t` has type `X`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:304:25
   |
LL |     if let &Either::One(_t) = &e { }
   |                         ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:308:34
   |
LL |     while let &Either::One(_t) = &e { }
   |               ----------------   ^^ cannot move out of borrowed content
   |               |            |
   |               |            data moved here
   |               help: consider removing the `&`: `Either::One(_t)`
   |
note: move occurs because `_t` has type `X`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:308:28
   |
LL |     while let &Either::One(_t) = &e { }
   |                            ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:312:11
   |
LL |     match &e {
   |           ^^ cannot move out of borrowed content
LL |         //~^ ERROR cannot move
LL |         &Either::One(_t)
   |         ----------------
   |         |            |
   |         |            data moved here
   |         help: consider removing the `&`: `Either::One(_t)`
   |
note: move occurs because `_t` has type `X`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:314:22
   |
LL |         &Either::One(_t)
   |                      ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:320:11
   |
LL |     match &e {
   |           ^^ cannot move out of borrowed content
LL |         //~^ ERROR cannot move
LL |         &Either::One(_t) => (),
   |         ----------------
   |         |            |
   |         |            data moved here
   |         help: consider removing the `&`: `Either::One(_t)`
   |
note: move occurs because `_t` has type `X`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:322:22
   |
LL |         &Either::One(_t) => (),
   |                      ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:327:11
   |
LL |     match &e {
   |           ^^ cannot move out of borrowed content
LL |         //~^ ERROR cannot move
LL |         &Either::One(_t) => (),
   |         ----------------
   |         |            |
   |         |            data moved here
   |         help: consider removing the `&`: `Either::One(_t)`
   |
note: move occurs because `_t` has type `X`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:329:22
   |
LL |         &Either::One(_t) => (),
   |                      ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:335:22
   |
LL |     let &mut X(_t) = &mut xm;
   |         ----------   ^^^^^^^ cannot move out of borrowed content
   |         |      |
   |         |      data moved here
   |         help: consider removing the `&mut`: `X(_t)`
   |
note: move occurs because `_t` has type `Y`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:335:16
   |
LL |     let &mut X(_t) = &mut xm;
   |                ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:339:35
   |
LL |     if let &mut Either::One(_t) = &mut em { }
   |            --------------------   ^^^^^^^ cannot move out of borrowed content
   |            |                |
   |            |                data moved here
   |            help: consider removing the `&mut`: `Either::One(_t)`
   |
note: move occurs because `_t` has type `X`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:339:29
   |
LL |     if let &mut Either::One(_t) = &mut em { }
   |                             ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:343:38
   |
LL |     while let &mut Either::One(_t) = &mut em { }
   |               --------------------   ^^^^^^^ cannot move out of borrowed content
   |               |                |
   |               |                data moved here
   |               help: consider removing the `&mut`: `Either::One(_t)`
   |
note: move occurs because `_t` has type `X`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:343:32
   |
LL |     while let &mut Either::One(_t) = &mut em { }
   |                                ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:347:11
   |
LL |     match &mut em {
   |           ^^^^^^^ cannot move out of borrowed content
LL |         //~^ ERROR cannot move
LL |         &mut Either::One(_t)
   |         --------------------
   |         |                |
   |         |                data moved here
   |         help: consider removing the `&mut`: `Either::One(_t)`
   |
note: move occurs because `_t` has type `X`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:349:26
   |
LL |         &mut Either::One(_t)
   |                          ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:355:11
   |
LL |     match &mut em {
   |           ^^^^^^^ cannot move out of borrowed content
LL |         //~^ ERROR cannot move
LL |         &mut Either::One(_t) => (),
   |         --------------------
   |         |                |
   |         |                data moved here
   |         help: consider removing the `&mut`: `Either::One(_t)`
   |
note: move occurs because `_t` has type `X`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:357:26
   |
LL |         &mut Either::One(_t) => (),
   |                          ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:362:11
   |
LL |     match &mut em {
   |           ^^^^^^^ cannot move out of borrowed content
LL |         //~^ ERROR cannot move
LL |         &mut Either::One(_t) => (),
   |         --------------------
   |         |                |
   |         |                data moved here
   |         help: consider removing the `&mut`: `Either::One(_t)`
   |
note: move occurs because `_t` has type `X`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:364:26
   |
LL |         &mut Either::One(_t) => (),
   |                          ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:369:11
   |
LL |     match &mut em {
   |           ^^^^^^^ cannot move out of borrowed content
LL |         //~^ ERROR cannot move
LL |         &mut Either::One(_t) => (),
   |         --------------------
   |         |                |
   |         |                data moved here
   |         help: consider removing the `&mut`: `Either::One(_t)`
   |
note: move occurs because `_t` has type `X`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:371:26
   |
LL |         &mut Either::One(_t) => (),
   |                          ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:379:27
   |
LL |     let &(X(_t), X(_u)) = &(x.clone(), x.clone());
   |         ---------------   ^^^^^^^^^^^^^^^^^^^^^^^ cannot move out of borrowed content
   |         |   |      |
   |         |   |      ...and here
   |         |   data moved here
   |         help: consider removing the `&`: `(X(_t), X(_u))`
   |
note: move occurs because these variables have types that don't implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:379:13
   |
LL |     let &(X(_t), X(_u)) = &(x.clone(), x.clone());
   |             ^^     ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:383:50
   |
LL |     if let &(Either::One(_t), Either::Two(_u)) = &(e.clone(), e.clone()) { }
   |            -----------------------------------   ^^^^^^^^^^^^^^^^^^^^^^^ cannot move out of borrowed content
   |            |             |                |
   |            |             |                ...and here
   |            |             data moved here
   |            help: consider removing the `&`: `(Either::One(_t), Either::Two(_u))`
   |
note: move occurs because these variables have types that don't implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:383:26
   |
LL |     if let &(Either::One(_t), Either::Two(_u)) = &(e.clone(), e.clone()) { }
   |                          ^^               ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:387:53
   |
LL |     while let &(Either::One(_t), Either::Two(_u)) = &(e.clone(), e.clone()) { }
   |               -----------------------------------   ^^^^^^^^^^^^^^^^^^^^^^^ cannot move out of borrowed content
   |               |             |                |
   |               |             |                ...and here
   |               |             data moved here
   |               help: consider removing the `&`: `(Either::One(_t), Either::Two(_u))`
   |
note: move occurs because these variables have types that don't implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:387:29
   |
LL |     while let &(Either::One(_t), Either::Two(_u)) = &(e.clone(), e.clone()) { }
   |                             ^^               ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:391:11
   |
LL |     match &(e.clone(), e.clone()) {
   |           ^^^^^^^^^^^^^^^^^^^^^^^ cannot move out of borrowed content
LL |         //~^ ERROR cannot move
LL |         &(Either::One(_t), Either::Two(_u)) => (),
   |                       --               -- ...and here
   |                       |
   |                       data moved here
...
LL |         &(Either::Two(_t), Either::One(_u)) => (),
   |                       -- ...and here   -- ...and here
   |
note: move occurs because these variables have types that don't implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:393:23
   |
LL |         &(Either::One(_t), Either::Two(_u)) => (),
   |                       ^^               ^^
...
LL |         &(Either::Two(_t), Either::One(_u)) => (),
   |                       ^^               ^^
help: consider removing the `&`
   |
LL |         (Either::One(_t), Either::Two(_u)) => (),
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
help: consider removing the `&`
   |
LL |         (Either::Two(_t), Either::One(_u)) => (),
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:401:11
   |
LL |     match &(e.clone(), e.clone()) {
   |           ^^^^^^^^^^^^^^^^^^^^^^^ cannot move out of borrowed content
LL |         //~^ ERROR cannot move
LL |         &(Either::One(_t), Either::Two(_u))
   |         -----------------------------------
   |         |             |                |
   |         |             |                ...and here
   |         |             data moved here
   |         help: consider removing the `&`: `(Either::One(_t), Either::Two(_u))`
   |
note: move occurs because these variables have types that don't implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:403:23
   |
LL |         &(Either::One(_t), Either::Two(_u))
   |                       ^^               ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:410:11
   |
LL |     match &(e.clone(), e.clone()) {
   |           ^^^^^^^^^^^^^^^^^^^^^^^ cannot move out of borrowed content
LL |         //~^ ERROR cannot move
LL |         &(Either::One(_t), Either::Two(_u)) => (),
   |         -----------------------------------
   |         |             |                |
   |         |             |                ...and here
   |         |             data moved here
   |         help: consider removing the `&`: `(Either::One(_t), Either::Two(_u))`
   |
note: move occurs because these variables have types that don't implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:412:23
   |
LL |         &(Either::One(_t), Either::Two(_u)) => (),
   |                       ^^               ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:418:11
   |
LL |     match &(e.clone(), e.clone()) {
   |           ^^^^^^^^^^^^^^^^^^^^^^^ cannot move out of borrowed content
LL |         //~^ ERROR cannot move
LL |         &(Either::One(_t), Either::Two(_u)) => (),
   |         -----------------------------------
   |         |             |                |
   |         |             |                ...and here
   |         |             data moved here
   |         help: consider removing the `&`: `(Either::One(_t), Either::Two(_u))`
   |
note: move occurs because these variables have types that don't implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:420:23
   |
LL |         &(Either::One(_t), Either::Two(_u)) => (),
   |                       ^^               ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:431:31
   |
LL |     let &mut (X(_t), X(_u)) = &mut (xm.clone(), xm.clone());
   |         -------------------   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ cannot move out of borrowed content
   |         |       |      |
   |         |       |      ...and here
   |         |       data moved here
   |         help: consider removing the `&mut`: `(X(_t), X(_u))`
   |
note: move occurs because these variables have types that don't implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:431:17
   |
LL |     let &mut (X(_t), X(_u)) = &mut (xm.clone(), xm.clone());
   |                 ^^     ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:435:54
   |
LL |     if let &mut (Either::One(_t), Either::Two(_u)) = &mut (em.clone(), em.clone()) { }
   |            ---------------------------------------   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ cannot move out of borrowed content
   |            |                 |                |
   |            |                 |                ...and here
   |            |                 data moved here
   |            help: consider removing the `&mut`: `(Either::One(_t), Either::Two(_u))`
   |
note: move occurs because these variables have types that don't implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:435:30
   |
LL |     if let &mut (Either::One(_t), Either::Two(_u)) = &mut (em.clone(), em.clone()) { }
   |                              ^^               ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:439:57
   |
LL |     while let &mut (Either::One(_t), Either::Two(_u)) = &mut (em.clone(), em.clone()) { }
   |               ---------------------------------------   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ cannot move out of borrowed content
   |               |                 |                |
   |               |                 |                ...and here
   |               |                 data moved here
   |               help: consider removing the `&mut`: `(Either::One(_t), Either::Two(_u))`
   |
note: move occurs because these variables have types that don't implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:439:33
   |
LL |     while let &mut (Either::One(_t), Either::Two(_u)) = &mut (em.clone(), em.clone()) { }
   |                                 ^^               ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:443:11
   |
LL |     match &mut (em.clone(), em.clone()) {
   |           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ cannot move out of borrowed content
LL |         //~^ ERROR cannot move
LL |         &mut (Either::One(_t), Either::Two(_u)) => (),
   |                           --               -- ...and here
   |                           |
   |                           data moved here
...
LL |         &mut (Either::Two(_t), Either::One(_u)) => (),
   |                           -- ...and here   -- ...and here
   |
note: move occurs because these variables have types that don't implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:445:27
   |
LL |         &mut (Either::One(_t), Either::Two(_u)) => (),
   |                           ^^               ^^
...
LL |         &mut (Either::Two(_t), Either::One(_u)) => (),
   |                           ^^               ^^
help: consider removing the `&mut`
   |
LL |         (Either::One(_t), Either::Two(_u)) => (),
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
help: consider removing the `&mut`
   |
LL |         (Either::Two(_t), Either::One(_u)) => (),
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:453:11
   |
LL |     match &mut (em.clone(), em.clone()) {
   |           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ cannot move out of borrowed content
LL |         //~^ ERROR cannot move
LL |         &mut (Either::One(_t), Either::Two(_u))
   |         ---------------------------------------
   |         |                 |                |
   |         |                 |                ...and here
   |         |                 data moved here
   |         help: consider removing the `&mut`: `(Either::One(_t), Either::Two(_u))`
   |
note: move occurs because these variables have types that don't implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:455:27
   |
LL |         &mut (Either::One(_t), Either::Two(_u))
   |                           ^^               ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:462:11
   |
LL |     match &mut (em.clone(), em.clone()) {
   |           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ cannot move out of borrowed content
LL |         //~^ ERROR cannot move
LL |         &mut (Either::One(_t), Either::Two(_u)) => (),
   |         ---------------------------------------
   |         |                 |                |
   |         |                 |                ...and here
   |         |                 data moved here
   |         help: consider removing the `&mut`: `(Either::One(_t), Either::Two(_u))`
   |
note: move occurs because these variables have types that don't implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:464:27
   |
LL |         &mut (Either::One(_t), Either::Two(_u)) => (),
   |                           ^^               ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:470:11
   |
LL |     match &mut (em.clone(), em.clone()) {
   |           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ cannot move out of borrowed content
LL |         //~^ ERROR cannot move
LL |         &mut (Either::One(_t), Either::Two(_u)) => (),
   |         ---------------------------------------
   |         |                 |                |
   |         |                 |                ...and here
   |         |                 data moved here
   |         help: consider removing the `&mut`: `(Either::One(_t), Either::Two(_u))`
   |
note: move occurs because these variables have types that don't implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:472:27
   |
LL |         &mut (Either::One(_t), Either::Two(_u)) => (),
   |                           ^^               ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:478:11
   |
LL |     match &mut (em.clone(), em.clone()) {
   |           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ cannot move out of borrowed content
LL |         //~^ ERROR cannot move
LL |         &mut (Either::One(_t), Either::Two(_u)) => (),
   |         ---------------------------------------
   |         |                 |                |
   |         |                 |                ...and here
   |         |                 data moved here
   |         help: consider removing the `&mut`: `(Either::One(_t), Either::Two(_u))`
   |
note: move occurs because these variables have types that don't implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:480:27
   |
LL |         &mut (Either::One(_t), Either::Two(_u)) => (),
   |                           ^^               ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:214:11
   |
LL |     fn f1(&X(_t): &X) { }
   |           ^^^--^
   |           |  |
   |           |  data moved here
   |           cannot move out of borrowed content
   |           help: consider removing the `&`: `X(_t)`
   |
note: move occurs because `_t` has type `Y`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:214:14
   |
LL |     fn f1(&X(_t): &X) { }
   |              ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:261:11
   |
LL |     fn f2(&mut X(_t): &mut X) { }
   |           ^^^^^^^--^
   |           |      |
   |           |      data moved here
   |           cannot move out of borrowed content
   |           help: consider removing the `&mut`: `X(_t)`
   |
note: move occurs because `_t` has type `Y`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:261:18
   |
LL |     fn f2(&mut X(_t): &mut X) { }
   |                  ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:281:11
   |
LL |     fn f3((&X(_t),): (&X,)) { }
   |           ^^^^--^^^
   |           |   |
   |           |   data moved here
   |           cannot move out of borrowed content
   |
note: move occurs because `_t` has type `Y`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:281:15
   |
LL |     fn f3((&X(_t),): (&X,)) { }
   |               ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:295:11
   |
LL |     fn f4((&mut X(_t),): (&mut X,)) { }
   |           ^^^^^^^^--^^^
   |           |       |
   |           |       data moved here
   |           cannot move out of borrowed content
   |
note: move occurs because `_t` has type `Y`, which does not implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:295:19
   |
LL |     fn f4((&mut X(_t),): (&mut X,)) { }
   |                   ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:426:11
   |
LL |     fn f5(&(X(_t), X(_u)): &(X, X)) { }
   |           ^^^^--^^^^^--^^
   |           |   |      |
   |           |   |      ...and here
   |           |   data moved here
   |           cannot move out of borrowed content
   |           help: consider removing the `&`: `(X(_t), X(_u))`
   |
note: move occurs because these variables have types that don't implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:426:15
   |
LL |     fn f5(&(X(_t), X(_u)): &(X, X)) { }
   |               ^^     ^^

error[E0507]: cannot move out of borrowed content
  --> $DIR/dont-suggest-ref.rs:486:11
   |
LL |     fn f6(&mut (X(_t), X(_u)): &mut (X, X)) { }
   |           ^^^^^^^^--^^^^^--^^
   |           |       |      |
   |           |       |      ...and here
   |           |       data moved here
   |           cannot move out of borrowed content
   |           help: consider removing the `&mut`: `(X(_t), X(_u))`
   |
note: move occurs because these variables have types that don't implement the `Copy` trait
  --> $DIR/dont-suggest-ref.rs:486:19
   |
LL |     fn f6(&mut (X(_t), X(_u)): &mut (X, X)) { }
   |                   ^^     ^^

error: aborting due to 77 previous errors

For more information about this error, try `rustc --explain E0507`.
