error: unsatisfied lifetime constraints
  --> $DIR/closure-substs.rs:17:16
   |
LL | fn foo<'a>() {
   |        -- lifetime `'a` defined here
LL |     |x: &'a i32| -> &'static i32 {
LL |         return x; //~ ERROR
   |                ^ returning this value requires that `'a` must outlive `'static`

error: borrowed data escapes outside of closure
  --> $DIR/closure-substs.rs:23:9
   |
LL |     |x: &i32, b: fn(&'static i32)| {
   |      - `x` is a reference that is only valid in the closure body
LL |         b(x); //~ ERROR
   |         ^^^^ `x` escapes the closure body here

error: borrowed data escapes outside of closure
  --> $DIR/closure-substs.rs:23:9
   |
LL |     |x: &i32, b: fn(&'static i32)| {
   |      -        - `b` is declared here, outside of the closure body
   |      |
   |      `x` is a reference that is only valid in the closure body
LL |         b(x); //~ ERROR
   |         ^^^^ `x` escapes the closure body here

error: unsatisfied lifetime constraints
  --> $DIR/closure-substs.rs:23:9
   |
LL |     |x: &i32, b: fn(&'static i32)| {
   |     ------------------------------
   |     |   |
   |     |   let's call the lifetime of this reference `'1`
   |     lifetime `'2` represents this closure's body
LL |         b(x); //~ ERROR
   |         ^^^^ argument requires that `'1` must outlive `'2`
   |
   = note: closure implements `Fn`, so references to captured variables can't escape the closure

error: aborting due to 4 previous errors

