error[E0308]: mismatched types
  --> $DIR/break-while-condition.rs:3:11
   |
LL | fn main() {
   |           ^ expected !, found ()
   |
   = note: expected type `!`
              found type `()`

error[E0308]: mismatched types
  --> $DIR/break-while-condition.rs:16:13
   |
LL | /             while false { //~ ERROR mismatched types
LL | |                 break
LL | |             }
   | |_____________^ expected !, found ()
   |
   = note: expected type `!`
              found type `()`

error[E0308]: mismatched types
  --> $DIR/break-while-condition.rs:24:13
   |
LL | /             while false { //~ ERROR mismatched types
LL | |                 return
LL | |             }
   | |_____________^ expected !, found ()
   |
   = note: expected type `!`
              found type `()`

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0308`.
