<<<<<<< HEAD
warning: constant evaluation error: non-constant path in constant expression
  --> $DIR/const-fn-error.rs:27:19
   |
LL |     let a : [i32; f(X)];
   |                   ^^^^
   |
   = note: #[warn(const_err)] on by default

error[E0016]: blocks in constant functions are limited to items and tail expressions
  --> $DIR/const-fn-error.rs:16:19
   |
LL |     let mut sum = 0; //~ ERROR blocks in constant functions are limited
=======
error[E0016]: blocks in constant functions are limited to items and tail expressions
  --> $DIR/const-fn-error.rs:16:19
   |
16 |     let mut sum = 0;
>>>>>>> Produce  instead of pointers
   |                   ^

error[E0015]: calls in constant functions are limited to constant functions, struct and enum constructors
  --> $DIR/const-fn-error.rs:18:14
   |
<<<<<<< HEAD
LL |     for i in 0..x { //~ ERROR calls in constant functions
=======
18 |     for i in 0..x {
>>>>>>> Report errors in statics during collecting instead of translating
   |              ^^^^

error[E0019]: constant function contains unimplemented expression type
  --> $DIR/const-fn-error.rs:18:14
   |
<<<<<<< HEAD
LL |     for i in 0..x { //~ ERROR calls in constant functions
=======
18 |     for i in 0..x {
>>>>>>> Report errors in statics during collecting instead of translating
   |              ^^^^

error[E0080]: constant evaluation error
<<<<<<< HEAD
  --> $DIR/const-fn-error.rs:21:5
   |
LL |     sum //~ ERROR E0080
   |     ^^^ non-constant path in constant expression
   |
note: for constant expression here
  --> $DIR/const-fn-error.rs:27:13
   |
LL |     let a : [i32; f(X)];
   |             ^^^^^^^^^^^
=======
  --> $DIR/const-fn-error.rs:28:19
   |
28 |     let a : [i32; f(X)];
   |                   ^^^^ miri failed: machine error: Cannot evaluate within constants: "calling non-const fn `<I as std::iter::IntoIterator><std::ops::Range<usize>>::into_iter`"
>>>>>>> Produce  instead of pointers

error: aborting due to 4 previous errors

You've got a few errors: E0015, E0016, E0019, E0080
If you want more information on an error, try using "rustc --explain E0015"
