error: patterns aren't allowed in trait methods
  --> $DIR/E0642.rs:12:12
   |
LL |     fn foo((x, y): (i32, i32)); //~ ERROR patterns aren't allowed in trait methods
   |            ^^^^^^
help: give this argument a name or use an underscore to ignore it
   |
LL |     fn foo(_: (i32, i32)); //~ ERROR patterns aren't allowed in trait methods
   |            ^

error: patterns aren't allowed in trait methods
  --> $DIR/E0642.rs:16:12
   |
LL |     fn bar((x, y): (i32, i32)) {} //~ ERROR patterns aren't allowed in trait methods
   |            ^^^^^^
help: give this argument a name or use an underscore to ignore it
   |
LL |     fn bar(_: (i32, i32)) {} //~ ERROR patterns aren't allowed in trait methods
   |            ^

error: aborting due to 2 previous errors

