error[E0642]: patterns aren't allowed in trait methods
  --> $DIR/E0642.rs:12:12
   |
LL |     fn foo((x, y): (i32, i32)); //~ ERROR patterns aren't allowed in trait methods
   |            ^^^^^^
help: give this argument a name or use an underscore to ignore it instead of using a tuple pattern
   |
LL |     fn foo(_: (i32, i32)); //~ ERROR patterns aren't allowed in trait methods
   |            ^

error[E0642]: patterns aren't allowed in trait methods
  --> $DIR/E0642.rs:16:12
   |
LL |     fn bar((x, y): (i32, i32)) {} //~ ERROR patterns aren't allowed in trait methods
   |            ^^^^^^
help: give this argument a name or use an underscore to ignore it instead of using a tuple pattern
   |
LL |     fn bar(_: (i32, i32)) {} //~ ERROR patterns aren't allowed in trait methods
   |            ^

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0642`.
