error: expected identifier, found reserved keyword `async`
  --> $DIR/edition-keywords-2018-2018-parsing.rs:37:13
   |
LL |     let mut async = 1; //~ ERROR expected identifier, found reserved keyword `async`
   |             ^^^^^ expected identifier, found reserved keyword

error: expected identifier, found reserved keyword `async`
  --> $DIR/edition-keywords-2018-2018-parsing.rs:47:13
   |
LL |     module::async(); //~ ERROR expected identifier, found reserved keyword `async`
   |             ^^^^^ expected identifier, found reserved keyword

error: no rules expected the token `r#proc`
  --> $DIR/edition-keywords-2018-2018-parsing.rs:25:29
   |
LL |     r#proc = consumes_proc!(r#proc); //~ ERROR no rules expected the token `r#proc`
   |                             ^^^^^^

error: no rules expected the token `proc`
  --> $DIR/edition-keywords-2018-2018-parsing.rs:26:33
   |
LL |     r#proc = consumes_proc_raw!(proc); //~ ERROR no rules expected the token `proc`
   |                                 ^^^^

error: no rules expected the token `r#async`
  --> $DIR/edition-keywords-2018-2018-parsing.rs:41:31
   |
LL |     r#async = consumes_async!(r#async); //~ ERROR no rules expected the token `r#async`
   |                               ^^^^^^^

error: no rules expected the token `async`
  --> $DIR/edition-keywords-2018-2018-parsing.rs:42:35
   |
LL |     r#async = consumes_async_raw!(async); //~ ERROR no rules expected the token `async`
   |                                   ^^^^^

error: expected expression, found reserved keyword `async`
  --> $DIR/edition-keywords-2018-2018-parsing.rs:45:22
   |
LL |     if passes_ident!(async) == 1 {} //~ ERROR expected expression, found reserved keyword `async`
   |                      ^^^^^ expected expression

error: aborting due to 7 previous errors

