error: unreachable pattern
  --> $DIR/exhaustive_integer_patterns.rs:32:9
   |
LL |         200 => {} //~ ERROR unreachable pattern
   |         ^^^
   |
note: lint level defined here
  --> $DIR/exhaustive_integer_patterns.rs:13:9
   |
LL | #![deny(unreachable_patterns)]
   |         ^^^^^^^^^^^^^^^^^^^^

error[E0004]: non-exhaustive patterns: `128u8...255u8` not covered
  --> $DIR/exhaustive_integer_patterns.rs:37:11
   |
LL |     match x { //~ ERROR non-exhaustive patterns
   |           ^ pattern `128u8...255u8` not covered

error[E0004]: non-exhaustive patterns: `11u8...19u8`, `31u8...34u8`, `36u8...69u8` and 1 more not covered
  --> $DIR/exhaustive_integer_patterns.rs:42:11
   |
LL |     match x { //~ ERROR non-exhaustive patterns
   |           ^ patterns `11u8...19u8`, `31u8...34u8`, `36u8...69u8` and 1 more not covered

error: unreachable pattern
  --> $DIR/exhaustive_integer_patterns.rs:53:9
   |
LL |         -2..=20 => {} //~ ERROR unreachable pattern
   |         ^^^^^^^

error[E0004]: non-exhaustive patterns: `-128i8...-8i8`, `-6i8`, `121i8...124i8` and 1 more not covered
  --> $DIR/exhaustive_integer_patterns.rs:50:11
   |
LL |     match x { //~ ERROR non-exhaustive patterns
   |           ^ patterns `-128i8...-8i8`, `-6i8`, `121i8...124i8` and 1 more not covered

error[E0004]: non-exhaustive patterns: `-128i8` not covered
  --> $DIR/exhaustive_integer_patterns.rs:99:11
   |
LL |     match 0i8 { //~ ERROR non-exhaustive patterns
   |           ^^^ pattern `-128i8` not covered

error[E0004]: non-exhaustive patterns: `0i16` not covered
  --> $DIR/exhaustive_integer_patterns.rs:107:11
   |
LL |     match 0i16 { //~ ERROR non-exhaustive patterns
   |           ^^^^ pattern `0i16` not covered

error: aborting due to 7 previous errors

For more information about this error, try `rustc --explain E0004`.
