error[E0378]: the trait `CoerceSized` may only be implemented for structs containing the field being coerced, `PhantomData` fields, and nothing else
  --> $DIR/invalid_coerce_sized_impls.rs:25:1
   |
LL | / impl<T, U> CoerceSized<WrapperWithExtraField<T>> for WrapperWithExtraField<U>
LL | | where
LL | |     T: CoerceUnsized<U>,
LL | |     U: CoerceSized<T>,
LL | | {} //~^^^^ ERROR [E0378]
   | |__^
   |
   = note: extra field `1` of type `i32` is not allowed

error[E0378]: implementing the `CoerceSized` trait requires multiple coercions
  --> $DIR/invalid_coerce_sized_impls.rs:39:1
   |
LL | / impl<T: ?Sized, U: ?Sized> CoerceSized<MultiplePointers<T>> for MultiplePointers<U>
LL | | where
LL | |     T: Unsize<U>,
LL | | {} //~^^^ ERROR [E0378]
   | |__^
   |
   = note: the trait `CoerceSized` may only be implemented for a coercion between structures with a single field being coerced
   = note: currently, 2 fields need coercions: ptr1 (*const U to *const T), ptr2 (*const U to *const T)

error[E0378]: the trait `CoerceSized` may only be implemented for a coercion between structures with a single field being coerced, none found
  --> $DIR/invalid_coerce_sized_impls.rs:51:1
   |
LL | impl<T: ?Sized, U: ?Sized> CoerceSized<NothingToCoerce<U>> for NothingToCoerce<T> {}
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0378`.
