error[E0265]: recursive constant
  --> $DIR/issue-36163.rs:11:1
   |
LL | const A: i32 = Foo::B; //~ ERROR E0265
   | ^^^^^^^^^^^^^^^^^^^^^^ recursion not allowed in constant

error[E0265]: recursive constant
  --> $DIR/issue-36163.rs:14:9
   |
LL |     B = A, //~ ERROR E0265
   |         ^ recursion not allowed in constant

error[E0265]: recursive constant
  --> $DIR/issue-36163.rs:18:9
   |
LL |     C = Bar::C, //~ ERROR E0265
   |         ^^^^^^ recursion not allowed in constant

error: aborting due to previous error

If you want more information on this error, try using "rustc --explain E0391"
