error[E0308]: mismatched types
  --> $DIR/liveness-return-last-stmt-semi.rs:13:45
   |
13 | macro_rules! test { () => { fn foo() -> i32 { 1; } } }
   |                                             ^^^-^^
   |                                             |  |
   |                                             |  help: consider removing this semicolon
   |                                             expected i32, found ()
...
28 |     test!();
   |     -------- in this macro invocation
   |
   = note: expected type `i32`
              found type `()`

error[E0308]: mismatched types
  --> $DIR/liveness-return-last-stmt-semi.rs:17:23
   |
17 | fn no_return() -> i32 {} //~ ERROR mismatched types
   |                       ^^ expected i32, found ()
   |
   = note: expected type `i32`
              found type `()`

error[E0308]: mismatched types
  --> $DIR/liveness-return-last-stmt-semi.rs:19:23
   |
19 |   fn bar(x: u32) -> u32 { //~ ERROR mismatched types
   |  _______________________^
20 | |     x * 2; //~ HELP consider removing this semicolon
   | |          - help: consider removing this semicolon
21 | | }
   | |_^ expected u32, found ()
   |
   = note: expected type `u32`
              found type `()`

error[E0308]: mismatched types
  --> $DIR/liveness-return-last-stmt-semi.rs:23:23
   |
23 |   fn baz(x: u64) -> u32 { //~ ERROR mismatched types
   |  _______________________^
24 | |     x * 2;
25 | | }
   | |_^ expected u32, found ()
   |
   = note: expected type `u32`
              found type `()`

error: aborting due to 4 previous errors

