warning: label name `'fl` shadows a label name that is already in scope
  --> $DIR/loops-reject-duplicate-labels-2.rs:23:7
   |
22 |     { 'fl: for _ in 0..10 { break; } }   //~ NOTE first declared here
   |       --- first declared here
23 |     { 'fl: loop { break; } }             //~ WARN label name `'fl` shadows a label name that is already in scope
   |       ^^^ lifetime 'fl already in scope

warning: label name `'lf` shadows a label name that is already in scope
  --> $DIR/loops-reject-duplicate-labels-2.rs:26:7
   |
25 |     { 'lf: loop { break; } }             //~ NOTE first declared here
   |       --- first declared here
26 |     { 'lf: for _ in 0..10 { break; } }   //~ WARN label name `'lf` shadows a label name that is already in scope
   |       ^^^ lifetime 'lf already in scope

warning: label name `'wl` shadows a label name that is already in scope
  --> $DIR/loops-reject-duplicate-labels-2.rs:29:7
   |
28 |     { 'wl: while 2 > 1 { break; } }      //~ NOTE first declared here
   |       --- first declared here
29 |     { 'wl: loop { break; } }             //~ WARN label name `'wl` shadows a label name that is already in scope
   |       ^^^ lifetime 'wl already in scope

warning: label name `'lw` shadows a label name that is already in scope
  --> $DIR/loops-reject-duplicate-labels-2.rs:32:7
   |
31 |     { 'lw: loop { break; } }             //~ NOTE first declared here
   |       --- first declared here
32 |     { 'lw: while 2 > 1 { break; } }      //~ WARN label name `'lw` shadows a label name that is already in scope
   |       ^^^ lifetime 'lw already in scope

warning: label name `'fw` shadows a label name that is already in scope
  --> $DIR/loops-reject-duplicate-labels-2.rs:35:7
   |
34 |     { 'fw: for _ in 0..10 { break; } }   //~ NOTE first declared here
   |       --- first declared here
35 |     { 'fw: while 2 > 1 { break; } }      //~ WARN label name `'fw` shadows a label name that is already in scope
   |       ^^^ lifetime 'fw already in scope

warning: label name `'wf` shadows a label name that is already in scope
  --> $DIR/loops-reject-duplicate-labels-2.rs:38:7
   |
37 |     { 'wf: while 2 > 1 { break; } }      //~ NOTE first declared here
   |       --- first declared here
38 |     { 'wf: for _ in 0..10 { break; } }   //~ WARN label name `'wf` shadows a label name that is already in scope
   |       ^^^ lifetime 'wf already in scope

warning: label name `'tl` shadows a label name that is already in scope
  --> $DIR/loops-reject-duplicate-labels-2.rs:41:7
   |
40 |     { 'tl: while let Some(_) = None::<i32> { break; } } //~ NOTE first declared here
   |       --- first declared here
41 |     { 'tl: loop { break; } }             //~ WARN label name `'tl` shadows a label name that is already in scope
   |       ^^^ lifetime 'tl already in scope

warning: label name `'lt` shadows a label name that is already in scope
  --> $DIR/loops-reject-duplicate-labels-2.rs:44:7
   |
43 |     { 'lt: loop { break; } }             //~ NOTE first declared here
   |       --- first declared here
44 |     { 'lt: while let Some(_) = None::<i32> { break; } }
   |       ^^^ lifetime 'lt already in scope

error: compilation successful
  --> $DIR/loops-reject-duplicate-labels-2.rs:50:1
   |
50 | / pub fn main() { //~ ERROR compilation successful
51 | |     foo();
52 | | }
   | |_^

