warning: label name `'fl` shadows a label name that is already in scope
  --> $DIR/loops-reject-duplicate-labels.rs:20:5
   |
19 |     'fl: for _ in 0..10 { break; } //~ NOTE first declared here
   |     --- first declared here
20 |     'fl: loop { break; }           //~ WARN label name `'fl` shadows a label name that is already in scope
   |     ^^^ lifetime 'fl already in scope

warning: label name `'lf` shadows a label name that is already in scope
  --> $DIR/loops-reject-duplicate-labels.rs:24:5
   |
23 |     'lf: loop { break; }           //~ NOTE first declared here
   |     --- first declared here
24 |     'lf: for _ in 0..10 { break; } //~ WARN label name `'lf` shadows a label name that is already in scope
   |     ^^^ lifetime 'lf already in scope

warning: label name `'wl` shadows a label name that is already in scope
  --> $DIR/loops-reject-duplicate-labels.rs:27:5
   |
26 |     'wl: while 2 > 1 { break; }    //~ NOTE first declared here
   |     --- first declared here
27 |     'wl: loop { break; }           //~ WARN label name `'wl` shadows a label name that is already in scope
   |     ^^^ lifetime 'wl already in scope

warning: label name `'lw` shadows a label name that is already in scope
  --> $DIR/loops-reject-duplicate-labels.rs:30:5
   |
29 |     'lw: loop { break; }           //~ NOTE first declared here
   |     --- first declared here
30 |     'lw: while 2 > 1 { break; }    //~ WARN label name `'lw` shadows a label name that is already in scope
   |     ^^^ lifetime 'lw already in scope

warning: label name `'fw` shadows a label name that is already in scope
  --> $DIR/loops-reject-duplicate-labels.rs:33:5
   |
32 |     'fw: for _ in 0..10 { break; } //~ NOTE first declared here
   |     --- first declared here
33 |     'fw: while 2 > 1 { break; }    //~ WARN label name `'fw` shadows a label name that is already in scope
   |     ^^^ lifetime 'fw already in scope

warning: label name `'wf` shadows a label name that is already in scope
  --> $DIR/loops-reject-duplicate-labels.rs:36:5
   |
35 |     'wf: while 2 > 1 { break; }    //~ NOTE first declared here
   |     --- first declared here
36 |     'wf: for _ in 0..10 { break; } //~ WARN label name `'wf` shadows a label name that is already in scope
   |     ^^^ lifetime 'wf already in scope

warning: label name `'tl` shadows a label name that is already in scope
  --> $DIR/loops-reject-duplicate-labels.rs:39:5
   |
38 |     'tl: while let Some(_) = None::<i32> { break; } //~ NOTE first declared here
   |     --- first declared here
39 |     'tl: loop { break; }           //~ WARN label name `'tl` shadows a label name that is already in scope
   |     ^^^ lifetime 'tl already in scope

warning: label name `'lt` shadows a label name that is already in scope
  --> $DIR/loops-reject-duplicate-labels.rs:42:5
   |
41 |     'lt: loop { break; }           //~ NOTE first declared here
   |     --- first declared here
42 |     'lt: while let Some(_) = None::<i32> { break; }
   |     ^^^ lifetime 'lt already in scope

error: compilation successful
  --> $DIR/loops-reject-duplicate-labels.rs:57:1
   |
57 | / pub fn main() { //~ ERROR compilation successful
58 | |     let s = S;
59 | |     s.m1();
60 | |     s.m2();
61 | |     foo();
62 | | }
   | |_^

