error[E0308]: mismatched types
  --> $DIR/struct-path-self-type-mismatch.rs:17:23
   |
17 |         Self { inner: 1.5f32 }; //~ ERROR mismatched types
   |                       ^^^^^^ expected i32, found f32

error[E0308]: mismatched types
  --> $DIR/struct-path-self-type-mismatch.rs:28:20
   |
28 |             inner: u
   |                    ^ expected type parameter, found a different type parameter
   |
   = note: expected type `T`
              found type `U`

error[E0308]: mismatched types
  --> $DIR/struct-path-self-type-mismatch.rs:24:9
   |
23 |       fn new<U>(u: U) -> Foo<U> { //~ NOTE expected `Foo<U>` because of return type
   |                          ------ expected `Foo<U>` because of return type
24 | /         Self {
25 | |         //~^ ERROR mismatched types
26 | |         //~| NOTE expected type parameter, found a different type parameter
27 | |         //~| NOTE expected type `Foo<U>`
...  |
31 | |             //~| NOTE expected type `T`
32 | |         }
   | |_________^ expected type parameter, found a different type parameter
   |
   = note: expected type `Foo<U>`
              found type `Foo<T>`

error: aborting due to 3 previous errors

