error[E0645]: trait aliases are not yet implemented (see issue #41517)
  --> $DIR/trait-alias.rs:11:1
   |
11 | trait SimpleAlias = Default; //~ERROR E0645
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0645]: trait aliases are not yet implemented (see issue #41517)
  --> $DIR/trait-alias.rs:12:1
   |
12 | trait GenericAlias<T> = Iterator<Item=T>; //~ERROR E0645
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0645]: trait aliases are not yet implemented (see issue #41517)
  --> $DIR/trait-alias.rs:13:1
   |
13 | trait Partial<T> = IntoIterator<Item=T>; //~ERROR E0645
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0645]: trait aliases are not yet implemented (see issue #41517)
  --> $DIR/trait-alias.rs:22:1
   |
22 | trait WithWhere<Art, Thou> = Romeo + Romeo where Fore<(Art, Thou)>: Romeo; //~ERROR E0645
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0645]: trait aliases are not yet implemented (see issue #41517)
  --> $DIR/trait-alias.rs:23:1
   |
23 | trait BareWhere<Wild, Are> = where The<Wild>: Things<Are>; //~ERROR E0645
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0645]: trait aliases are not yet implemented (see issue #41517)
  --> $DIR/trait-alias.rs:25:1
   |
25 | trait CD = Clone + Default; //~ERROR E0645
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 6 previous errors

