error: doc comment not used by rustdoc
  --> $DIR/useless_comment.rs:4:5
   |
LL |     /// a //~ ERROR doc comment not used by rustdoc
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: lint level defined here
  --> $DIR/useless_comment.rs:1:9
   |
LL | #![deny(unused_doc_comments)]
   |         ^^^^^^^^^^^^^^^^^^^

error: doc comment not used by rustdoc
  --> $DIR/useless_comment.rs:7:5
   |
LL | /     /// multi-line //~ doc comment not used by rustdoc
LL | |     /// doc comment
LL | |     /// that is unused
   | |______________________^

error: doc comment not used by rustdoc
  --> $DIR/useless_comment.rs:11:9
   |
LL |         /// c //~ ERROR doc comment not used by rustdoc
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: doc comment not used by rustdoc
  --> $DIR/useless_comment.rs:16:5
   |
LL |     /// foo //~ ERROR doc comment not used by rustdoc
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: doc comment not used by rustdoc
  --> $DIR/useless_comment.rs:19:5
   |
LL |     #[doc = "foo"] //~ ERROR doc comment not used by rustdoc
   |     ^^^^^^^^^^^^^^

error: doc comment not used by rustdoc
  --> $DIR/useless_comment.rs:20:5
   |
LL |     #[doc = "bar"] //~ ERROR doc comment not used by rustdoc
   |     ^^^^^^^^^^^^^^

error: aborting due to 6 previous errors

