warning: not reporting region error due to -Znll
  --> $DIR/propagate-from-trait-match.rs:53:9
   |
53 |         require(value);
   |         ^^^^^^^

note: External requirements
  --> $DIR/propagate-from-trait-match.rs:42:36
   |
42 |       establish_relationships(value, |value| {
   |  ____________________________________^
43 | |         // This function call requires that
44 | |         //
45 | |         // (a) T: Trait<'a>
...  |
55 | |         //~| ERROR failed type test
56 | |     });
   | |_____^
   |
   = note: defining type: DefId(0/1:16 ~ propagate_from_trait_match[317d]::supply[0]::{{closure}}[0]) with closure substs [
               '_#1r,
               T,
               i32,
               extern "rust-call" fn((T,))
           ]
   = note: number of external vids: 2
   = note: where T: '_#1r

error: failed type test: TypeTest { generic_kind: T/#1, lower_bound: '_#3r, point: bb0[3], span: $DIR/propagate-from-trait-match.rs:42:36: 56:6, test: IsOutlivedByAnyRegionIn(['_#2r]) }
  --> $DIR/propagate-from-trait-match.rs:42:36
   |
42 |       establish_relationships(value, |value| {
   |  ____________________________________^
43 | |         // This function call requires that
44 | |         //
45 | |         // (a) T: Trait<'a>
...  |
55 | |         //~| ERROR failed type test
56 | |     });
   | |_____^

note: No external requirements
  --> $DIR/propagate-from-trait-match.rs:38:1
   |
38 | / fn supply<'a, T>(value: T)
39 | | where
40 | |     T: Trait<'a>,
41 | | {
...  |
56 | |     });
57 | | }
   | |_^
   |
   = note: defining type: DefId(0/0:6 ~ propagate_from_trait_match[317d]::supply[0]) with substs [
               '_#1r,
               T
           ]

error: aborting due to previous error

