error[E0087]: wrong number of type arguments: expected 0, found 1
  --> $DIR/E0087.rs:15:5
   |
LL |     foo::<f64>(); //~ ERROR wrong number of type arguments: expected 0, found 1 [E0087]
   |     ^^^^^^^^^^ unexpected type argument

error[E0087]: wrong number of type arguments: expected 1, found 2
  --> $DIR/E0087.rs:17:5
   |
LL |     bar::<f64, u64>(); //~ ERROR wrong number of type arguments: expected 1, found 2 [E0087]
   |     ^^^^^^^^^^^^^^^ unexpected type argument

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0087`.
