error: bitshift exceeds the type's number of bits
  --> $DIR/E0080.rs:12:9
   |
LL |     X = (1 << 500), //~ ERROR E0080
   |         ^^^^^^^^^^
   |
   = note: #[deny(exceeding_bitshifts)] on by default

error[E0080]: constant evaluation error
  --> $DIR/E0080.rs:12:9
   |
LL |     X = (1 << 500), //~ ERROR E0080
   |         ^^^^^^^^^^ attempt to shift left with overflow

warning: constant evaluation error
  --> $DIR/E0080.rs:15:9
   |
LL |     Y = (1 / 0) //~ ERROR E0080
   |         ^^^^^^^ attempt to divide by zero
   |
   = note: #[warn(const_err)] on by default

warning: constant evaluation error
  --> $DIR/E0080.rs:15:9
   |
LL |     Y = (1 / 0) //~ ERROR E0080
   |         ^^^^^^^ attempted to do overflowing math

error[E0080]: constant evaluation error
  --> $DIR/E0080.rs:15:9
   |
LL |     Y = (1 / 0) //~ ERROR E0080
   |         ^^^^^^^ attempt to divide by zero

error: aborting due to 3 previous errors

If you want more information on this error, try using "rustc --explain E0080"
