error[E0719]: the value of the associated type `Item` (from the trait `std::iter::Iterator`) is already specified
  --> $DIR/E0719-trait-alias.rs:14:34
   |
LL | trait I32Iterator = Iterator<Item = i32>;
   |                              ---------- `Item` bound here first
LL | trait I32Iterator2 = I32Iterator<Item = i32>; //~ ERROR E0719
   |                                  ^^^^^^^^^^ re-bound here

error[E0719]: the value of the associated type `Item` (from the trait `std::iter::Iterator`) is already specified
  --> $DIR/E0719-trait-alias.rs:15:34
   |
LL | trait I32Iterator = Iterator<Item = i32>;
   |                              ---------- `Item` bound here first
LL | trait I32Iterator2 = I32Iterator<Item = i32>; //~ ERROR E0719
LL | trait U32Iterator = I32Iterator2<Item = i32>; //~ ERROR E0719
   |                                  ^^^^^^^^^^ re-bound here

error[E0719]: the value of the associated type `Item` (from the trait `std::iter::Iterator`) is already specified
  --> $DIR/E0719-trait-alias.rs:16:34
   |
LL | trait I32Iterator = Iterator<Item = i32>;
   |                              ---------- `Item` bound here first
...
LL | trait U32Iterator2 = U32Iterator<Item = u32>; //~ ERROR E0719
   |                                  ^^^^^^^^^^ re-bound here

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0719`.
