error: approximate value of `f{32, 64}::consts::E` found. Consider using it directly
 --> $DIR/approx_const.rs:7:16
  |
7 |     let my_e = 2.7182; //~ERROR approximate value of `f{32, 64}::consts::E` found
  |                ^^^^^^
  |
note: lint level defined here
 --> $DIR/approx_const.rs:4:8
  |
4 | #[deny(approx_constant)]
  |        ^^^^^^^^^^^^^^^

error: approximate value of `f{32, 64}::consts::E` found. Consider using it directly
 --> $DIR/approx_const.rs:8:20
  |
8 |     let almost_e = 2.718; //~ERROR approximate value of `f{32, 64}::consts::E` found
  |                    ^^^^^

error: approximate value of `f{32, 64}::consts::FRAC_1_PI` found. Consider using it directly
  --> $DIR/approx_const.rs:11:24
   |
11 |     let my_1_frac_pi = 0.3183; //~ERROR approximate value of `f{32, 64}::consts::FRAC_1_PI` found
   |                        ^^^^^^

error: approximate value of `f{32, 64}::consts::FRAC_1_SQRT_2` found. Consider using it directly
  --> $DIR/approx_const.rs:14:28
   |
14 |     let my_frac_1_sqrt_2 = 0.70710678; //~ERROR approximate value of `f{32, 64}::consts::FRAC_1_SQRT_2` found
   |                            ^^^^^^^^^^

error: approximate value of `f{32, 64}::consts::FRAC_1_SQRT_2` found. Consider using it directly
  --> $DIR/approx_const.rs:15:32
   |
15 |     let almost_frac_1_sqrt_2 = 0.70711; //~ERROR approximate value of `f{32, 64}::consts::FRAC_1_SQRT_2` found
   |                                ^^^^^^^

error: approximate value of `f{32, 64}::consts::FRAC_2_PI` found. Consider using it directly
  --> $DIR/approx_const.rs:18:24
   |
18 |     let my_frac_2_pi = 0.63661977; //~ERROR approximate value of `f{32, 64}::consts::FRAC_2_PI` found
   |                        ^^^^^^^^^^

error: approximate value of `f{32, 64}::consts::FRAC_2_SQRT_PI` found. Consider using it directly
  --> $DIR/approx_const.rs:21:27
   |
21 |     let my_frac_2_sq_pi = 1.128379; //~ERROR approximate value of `f{32, 64}::consts::FRAC_2_SQRT_PI` found
   |                           ^^^^^^^^

error: approximate value of `f{32, 64}::consts::FRAC_PI_2` found. Consider using it directly
  --> $DIR/approx_const.rs:24:24
   |
24 |     let my_frac_pi_2 = 1.57079632679; //~ERROR approximate value of `f{32, 64}::consts::FRAC_PI_2` found
   |                        ^^^^^^^^^^^^^

error: approximate value of `f{32, 64}::consts::FRAC_PI_3` found. Consider using it directly
  --> $DIR/approx_const.rs:27:24
   |
27 |     let my_frac_pi_3 = 1.04719755119; //~ERROR approximate value of `f{32, 64}::consts::FRAC_PI_3` found
   |                        ^^^^^^^^^^^^^

error: approximate value of `f{32, 64}::consts::FRAC_PI_4` found. Consider using it directly
  --> $DIR/approx_const.rs:30:24
   |
30 |     let my_frac_pi_4 = 0.785398163397; //~ERROR approximate value of `f{32, 64}::consts::FRAC_PI_4` found
   |                        ^^^^^^^^^^^^^^

error: approximate value of `f{32, 64}::consts::FRAC_PI_6` found. Consider using it directly
  --> $DIR/approx_const.rs:33:24
   |
33 |     let my_frac_pi_6 = 0.523598775598; //~ERROR approximate value of `f{32, 64}::consts::FRAC_PI_6` found
   |                        ^^^^^^^^^^^^^^

error: approximate value of `f{32, 64}::consts::FRAC_PI_8` found. Consider using it directly
  --> $DIR/approx_const.rs:36:24
   |
36 |     let my_frac_pi_8 = 0.3926990816987; //~ERROR approximate value of `f{32, 64}::consts::FRAC_PI_8` found
   |                        ^^^^^^^^^^^^^^^

error: approximate value of `f{32, 64}::consts::LN_10` found. Consider using it directly
  --> $DIR/approx_const.rs:39:20
   |
39 |     let my_ln_10 = 2.302585092994046; //~ERROR approximate value of `f{32, 64}::consts::LN_10` found
   |                    ^^^^^^^^^^^^^^^^^

error: approximate value of `f{32, 64}::consts::LN_2` found. Consider using it directly
  --> $DIR/approx_const.rs:42:19
   |
42 |     let my_ln_2 = 0.6931471805599453; //~ERROR approximate value of `f{32, 64}::consts::LN_2` found
   |                   ^^^^^^^^^^^^^^^^^^

error: approximate value of `f{32, 64}::consts::LOG10_E` found. Consider using it directly
  --> $DIR/approx_const.rs:45:22
   |
45 |     let my_log10_e = 0.43429448190325182; //~ERROR approximate value of `f{32, 64}::consts::LOG10_E` found
   |                      ^^^^^^^^^^^^^^^^^^^

error: approximate value of `f{32, 64}::consts::LOG2_E` found. Consider using it directly
  --> $DIR/approx_const.rs:48:21
   |
48 |     let my_log2_e = 1.4426950408889634; //~ERROR approximate value of `f{32, 64}::consts::LOG2_E` found
   |                     ^^^^^^^^^^^^^^^^^^

error: approximate value of `f{32, 64}::consts::PI` found. Consider using it directly
  --> $DIR/approx_const.rs:51:17
   |
51 |     let my_pi = 3.1415; //~ERROR approximate value of `f{32, 64}::consts::PI` found
   |                 ^^^^^^

error: approximate value of `f{32, 64}::consts::PI` found. Consider using it directly
  --> $DIR/approx_const.rs:52:21
   |
52 |     let almost_pi = 3.14; //~ERROR approximate value of `f{32, 64}::consts::PI` found
   |                     ^^^^

error: approximate value of `f{32, 64}::consts::SQRT_2` found. Consider using it directly
  --> $DIR/approx_const.rs:55:18
   |
55 |     let my_sq2 = 1.4142; //~ERROR approximate value of `f{32, 64}::consts::SQRT_2` found
   |                  ^^^^^^

error: aborting due to 19 previous errors

