error: you have declared `#[inline(always)]` on `test_attr_lint`. This is usually a bad idea
 --> $DIR/attrs.rs:6:1
  |
6 | #[inline(always)] //~ERROR you have declared `#[inline(always)]` on `test_attr_lint`.
  | ^^^^^^^^^^^^^^^^^
  |
note: lint level defined here
 --> $DIR/attrs.rs:4:9
  |
4 | #![deny(inline_always, deprecated_semver)]
  |         ^^^^^^^^^^^^^

error: the since field must contain a semver-compliant version
  --> $DIR/attrs.rs:27:14
   |
27 | #[deprecated(since = "forever")] //~ERROR the since field must contain a semver-compliant version
   |              ^^^^^^^^^^^^^^^^^
   |
note: lint level defined here
  --> $DIR/attrs.rs:4:24
   |
4  | #![deny(inline_always, deprecated_semver)]
   |                        ^^^^^^^^^^^^^^^^^

error: the since field must contain a semver-compliant version
  --> $DIR/attrs.rs:30:14
   |
30 | #[deprecated(since = "1")] //~ERROR the since field must contain a semver-compliant version
   |              ^^^^^^^^^^^

error: aborting due to 3 previous errors

