error: in an 'if' condition, avoid complex blocks or closures with blocks; instead, move the block or closure higher and bind it with a 'let'
  --> $DIR/block_in_if_condition.rs:30:8
   |
30 |       if { //~ERROR in an 'if' condition, avoid complex blocks or closures with blocks;
   |  ________^ starting here...
31 | |         let x = 3;
32 | |         x == 3
33 | |     } {
   | |_____^ ...ending here
   |
note: lint level defined here
  --> $DIR/block_in_if_condition.rs:5:9
   |
5  | #![deny(block_in_if_condition_stmt)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^
   = help: try
           let res = { //~ERROR in an 'if' condition, avoid complex blocks or closures with blocks;
               let x = 3;
               x == 3
           };
           if res {
               6
           } ... 

error: omit braces around single expression condition
  --> $DIR/block_in_if_condition.rs:41:8
   |
41 |     if { true } { //~ERROR omit braces around single expression condition
   |        ^^^^^^^^
   |
note: lint level defined here
  --> $DIR/block_in_if_condition.rs:4:9
   |
4  | #![deny(block_in_if_condition_expr)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^
   = help: try
           if true { //~ERROR omit braces around single expression condition
               6
           } ... 

error: in an 'if' condition, avoid complex blocks or closures with blocks; instead, move the block or closure higher and bind it with a 'let'
  --> $DIR/block_in_if_condition.rs:58:49
   |
58 |     if v == 3 && sky == "blue" && predicate(|x| { let target = 3; x == target }, v) {
   |                                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: in an 'if' condition, avoid complex blocks or closures with blocks; instead, move the block or closure higher and bind it with a 'let'
  --> $DIR/block_in_if_condition.rs:62:22
   |
62 |     if predicate(|x| { let target = 3; x == target }, v) {
   |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: this boolean expression can be simplified
  --> $DIR/block_in_if_condition.rs:70:8
   |
70 |     if true && x == 3 { //~ WARN this boolean expression can be simplified
   |        ^^^^^^^^^^^^^^
   |
note: lint level defined here
  --> $DIR/block_in_if_condition.rs:7:9
   |
7  | #![warn(nonminimal_bool)]
   |         ^^^^^^^^^^^^^^^
help: try
   |     if x == 3 { //~ WARN this boolean expression can be simplified

error: aborting due to 4 previous errors

