error: Comparing with null is better expressed by the .is_null() method
  --> $DIR/cmp_null.rs:11:8
   |
11 |     if p == ptr::null() { //~ERROR:  Comparing with null
   |        ^^^^^^^^^^^^^^^^
   |
note: lint level defined here
  --> $DIR/cmp_null.rs:3:9
   |
3  | #![deny(cmp_null)]
   |         ^^^^^^^^

error: Comparing with null is better expressed by the .is_null() method
  --> $DIR/cmp_null.rs:16:8
   |
16 |     if m == ptr::null_mut() { //~ERROR:  Comparing with null
   |        ^^^^^^^^^^^^^^^^^^^^

error: aborting due to 2 previous errors

