error: sub-expression diverges
  --> $DIR/diverging_sub_expression.rs:18:10
   |
18 |     b || diverge(); //~ ERROR sub-expression diverges
   |          ^^^^^^^^^
   |
note: lint level defined here
  --> $DIR/diverging_sub_expression.rs:3:9
   |
3  | #![deny(diverging_sub_expression)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^

error: sub-expression diverges
  --> $DIR/diverging_sub_expression.rs:19:10
   |
19 |     b || A.foo(); //~ ERROR sub-expression diverges
   |          ^^^^^^^

error: sub-expression diverges
  --> $DIR/diverging_sub_expression.rs:28:26
   |
28 |             6 => true || return, //~ ERROR sub-expression diverges
   |                          ^^^^^^

error: sub-expression diverges
  --> $DIR/diverging_sub_expression.rs:29:26
   |
29 |             7 => true || continue, //~ ERROR sub-expression diverges
   |                          ^^^^^^^^

error: sub-expression diverges
  --> $DIR/diverging_sub_expression.rs:32:26
   |
32 |             3 => true || diverge(), //~ ERROR sub-expression diverges
   |                          ^^^^^^^^^

error: sub-expression diverges
  --> $DIR/diverging_sub_expression.rs:37:26
   |
37 |             _ => true || break, //~ ERROR sub-expression diverges
   |                          ^^^^^

error: aborting due to 6 previous errors

