error: Consider removing unnecessary double parentheses
  --> $DIR/double_parens.rs:16:5
   |
16 |     ((0)) //~ERROR Consider removing unnecessary double parentheses
   |     ^^^^^
   |
note: lint level defined here
  --> $DIR/double_parens.rs:4:9
   |
4  | #![deny(double_parens)]
   |         ^^^^^^^^^^^^^

error: Consider removing unnecessary double parentheses
  --> $DIR/double_parens.rs:20:14
   |
20 |     dummy_fn((0)); //~ERROR Consider removing unnecessary double parentheses
   |              ^^^

error: Consider removing unnecessary double parentheses
  --> $DIR/double_parens.rs:24:20
   |
24 |     x.dummy_method((0)); //~ERROR Consider removing unnecessary double parentheses
   |                    ^^^

error: Consider removing unnecessary double parentheses
  --> $DIR/double_parens.rs:28:5
   |
28 |     ((1, 2)) //~ERROR Consider removing unnecessary double parentheses
   |     ^^^^^^^^

error: Consider removing unnecessary double parentheses
  --> $DIR/double_parens.rs:32:5
   |
32 |     (()) //~ERROR Consider removing unnecessary double parentheses
   |     ^^^^

error: aborting due to 5 previous errors

