error: call to `std::mem::drop` with a reference argument. Dropping a reference does nothing
  --> $DIR/drop_forget_ref.rs:12:5
   |
12 |     drop(&SomeStruct); //~ERROR call to `std::mem::drop` with a reference argument
   |     ^^^^^^^^^^^^^^^^^
   |
note: lint level defined here
  --> $DIR/drop_forget_ref.rs:4:9
   |
4  | #![deny(drop_ref, forget_ref)]
   |         ^^^^^^^^
note: argument has type &SomeStruct
  --> $DIR/drop_forget_ref.rs:12:10
   |
12 |     drop(&SomeStruct); //~ERROR call to `std::mem::drop` with a reference argument
   |          ^^^^^^^^^^^

error: call to `std::mem::forget` with a reference argument. Forgetting a reference does nothing
  --> $DIR/drop_forget_ref.rs:13:5
   |
13 |     forget(&SomeStruct); //~ERROR call to `std::mem::forget` with a reference argument
   |     ^^^^^^^^^^^^^^^^^^^
   |
note: lint level defined here
  --> $DIR/drop_forget_ref.rs:4:19
   |
4  | #![deny(drop_ref, forget_ref)]
   |                   ^^^^^^^^^^
note: argument has type &SomeStruct
  --> $DIR/drop_forget_ref.rs:13:12
   |
13 |     forget(&SomeStruct); //~ERROR call to `std::mem::forget` with a reference argument
   |            ^^^^^^^^^^^

error: call to `std::mem::drop` with a reference argument. Dropping a reference does nothing
  --> $DIR/drop_forget_ref.rs:16:5
   |
16 |     drop(&owned1); //~ERROR call to `std::mem::drop` with a reference argument
   |     ^^^^^^^^^^^^^
   |
note: argument has type &SomeStruct
  --> $DIR/drop_forget_ref.rs:16:10
   |
16 |     drop(&owned1); //~ERROR call to `std::mem::drop` with a reference argument
   |          ^^^^^^^

error: call to `std::mem::drop` with a reference argument. Dropping a reference does nothing
  --> $DIR/drop_forget_ref.rs:17:5
   |
17 |     drop(&&owned1); //~ERROR call to `std::mem::drop` with a reference argument
   |     ^^^^^^^^^^^^^^
   |
note: argument has type &&SomeStruct
  --> $DIR/drop_forget_ref.rs:17:10
   |
17 |     drop(&&owned1); //~ERROR call to `std::mem::drop` with a reference argument
   |          ^^^^^^^^

error: call to `std::mem::drop` with a reference argument. Dropping a reference does nothing
  --> $DIR/drop_forget_ref.rs:18:5
   |
18 |     drop(&mut owned1); //~ERROR call to `std::mem::drop` with a reference argument
   |     ^^^^^^^^^^^^^^^^^
   |
note: argument has type &mut SomeStruct
  --> $DIR/drop_forget_ref.rs:18:10
   |
18 |     drop(&mut owned1); //~ERROR call to `std::mem::drop` with a reference argument
   |          ^^^^^^^^^^^

error: call to `std::mem::forget` with a reference argument. Forgetting a reference does nothing
  --> $DIR/drop_forget_ref.rs:21:5
   |
21 |     forget(&owned2); //~ERROR call to `std::mem::forget` with a reference argument
   |     ^^^^^^^^^^^^^^^
   |
note: argument has type &SomeStruct
  --> $DIR/drop_forget_ref.rs:21:12
   |
21 |     forget(&owned2); //~ERROR call to `std::mem::forget` with a reference argument
   |            ^^^^^^^

error: call to `std::mem::forget` with a reference argument. Forgetting a reference does nothing
  --> $DIR/drop_forget_ref.rs:22:5
   |
22 |     forget(&&owned2); //~ERROR call to `std::mem::forget` with a reference argument
   |     ^^^^^^^^^^^^^^^^
   |
note: argument has type &&SomeStruct
  --> $DIR/drop_forget_ref.rs:22:12
   |
22 |     forget(&&owned2); //~ERROR call to `std::mem::forget` with a reference argument
   |            ^^^^^^^^

error: call to `std::mem::forget` with a reference argument. Forgetting a reference does nothing
  --> $DIR/drop_forget_ref.rs:23:5
   |
23 |     forget(&mut owned2); //~ERROR call to `std::mem::forget` with a reference argument
   |     ^^^^^^^^^^^^^^^^^^^
   |
note: argument has type &mut SomeStruct
  --> $DIR/drop_forget_ref.rs:23:12
   |
23 |     forget(&mut owned2); //~ERROR call to `std::mem::forget` with a reference argument
   |            ^^^^^^^^^^^

error: call to `std::mem::drop` with a reference argument. Dropping a reference does nothing
  --> $DIR/drop_forget_ref.rs:27:5
   |
27 |     drop(reference1); //~ERROR call to `std::mem::drop` with a reference argument
   |     ^^^^^^^^^^^^^^^^
   |
note: argument has type &SomeStruct
  --> $DIR/drop_forget_ref.rs:27:10
   |
27 |     drop(reference1); //~ERROR call to `std::mem::drop` with a reference argument
   |          ^^^^^^^^^^

error: call to `std::mem::forget` with a reference argument. Forgetting a reference does nothing
  --> $DIR/drop_forget_ref.rs:28:5
   |
28 |     forget(&*reference1); //~ERROR call to `std::mem::forget` with a reference argument
   |     ^^^^^^^^^^^^^^^^^^^^
   |
note: argument has type &SomeStruct
  --> $DIR/drop_forget_ref.rs:28:12
   |
28 |     forget(&*reference1); //~ERROR call to `std::mem::forget` with a reference argument
   |            ^^^^^^^^^^^^

error: call to `std::mem::drop` with a reference argument. Dropping a reference does nothing
  --> $DIR/drop_forget_ref.rs:31:5
   |
31 |     drop(reference2); //~ERROR call to `std::mem::drop` with a reference argument
   |     ^^^^^^^^^^^^^^^^
   |
note: argument has type &mut SomeStruct
  --> $DIR/drop_forget_ref.rs:31:10
   |
31 |     drop(reference2); //~ERROR call to `std::mem::drop` with a reference argument
   |          ^^^^^^^^^^

error: call to `std::mem::forget` with a reference argument. Forgetting a reference does nothing
  --> $DIR/drop_forget_ref.rs:33:5
   |
33 |     forget(reference3); //~ERROR call to `std::mem::forget` with a reference argument
   |     ^^^^^^^^^^^^^^^^^^
   |
note: argument has type &mut SomeStruct
  --> $DIR/drop_forget_ref.rs:33:12
   |
33 |     forget(reference3); //~ERROR call to `std::mem::forget` with a reference argument
   |            ^^^^^^^^^^

error: call to `std::mem::drop` with a reference argument. Dropping a reference does nothing
  --> $DIR/drop_forget_ref.rs:36:5
   |
36 |     drop(reference4); //~ERROR call to `std::mem::drop` with a reference argument
   |     ^^^^^^^^^^^^^^^^
   |
note: argument has type &SomeStruct
  --> $DIR/drop_forget_ref.rs:36:10
   |
36 |     drop(reference4); //~ERROR call to `std::mem::drop` with a reference argument
   |          ^^^^^^^^^^

error: call to `std::mem::forget` with a reference argument. Forgetting a reference does nothing
  --> $DIR/drop_forget_ref.rs:37:5
   |
37 |     forget(reference4); //~ERROR call to `std::mem::forget` with a reference argument
   |     ^^^^^^^^^^^^^^^^^^
   |
note: argument has type &SomeStruct
  --> $DIR/drop_forget_ref.rs:37:12
   |
37 |     forget(reference4); //~ERROR call to `std::mem::forget` with a reference argument
   |            ^^^^^^^^^^

error: call to `std::mem::drop` with a reference argument. Dropping a reference does nothing
  --> $DIR/drop_forget_ref.rs:42:5
   |
42 |     drop(&val); //~ERROR call to `std::mem::drop` with a reference argument
   |     ^^^^^^^^^^
   |
note: argument has type &T
  --> $DIR/drop_forget_ref.rs:42:10
   |
42 |     drop(&val); //~ERROR call to `std::mem::drop` with a reference argument
   |          ^^^^

error: call to `std::mem::forget` with a reference argument. Forgetting a reference does nothing
  --> $DIR/drop_forget_ref.rs:48:5
   |
48 |     forget(&val); //~ERROR call to `std::mem::forget` with a reference argument
   |     ^^^^^^^^^^^^
   |
note: argument has type &T
  --> $DIR/drop_forget_ref.rs:48:12
   |
48 |     forget(&val); //~ERROR call to `std::mem::forget` with a reference argument
   |            ^^^^

error: call to `std::mem::drop` with a reference argument. Dropping a reference does nothing
  --> $DIR/drop_forget_ref.rs:56:5
   |
56 |     std::mem::drop(&SomeStruct); //~ERROR call to `std::mem::drop` with a reference argument
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: argument has type &SomeStruct
  --> $DIR/drop_forget_ref.rs:56:20
   |
56 |     std::mem::drop(&SomeStruct); //~ERROR call to `std::mem::drop` with a reference argument
   |                    ^^^^^^^^^^^

error: call to `std::mem::forget` with a reference argument. Forgetting a reference does nothing
  --> $DIR/drop_forget_ref.rs:59:5
   |
59 |     std::mem::forget(&SomeStruct); //~ERROR call to `std::mem::forget` with a reference argument
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: argument has type &SomeStruct
  --> $DIR/drop_forget_ref.rs:59:22
   |
59 |     std::mem::forget(&SomeStruct); //~ERROR call to `std::mem::forget` with a reference argument
   |                      ^^^^^^^^^^^

error: aborting due to 18 previous errors

