error: enum with no variants
 --> $DIR/empty_enum.rs:7:1
  |
7 | enum Empty {} //~ ERROR enum with no variants
  | ^^^^^^^^^^^^^
  |
note: lint level defined here
 --> $DIR/empty_enum.rs:5:9
  |
5 | #![deny(empty_enum)]
  |         ^^^^^^^^^^
help: consider using the uninhabited type `!` or a wrapper around it
 --> $DIR/empty_enum.rs:7:1
  |
7 | enum Empty {} //~ ERROR enum with no variants
  | ^^^^^^^^^^^^^

error: aborting due to previous error

