error: Variant name ends with the enum's name
  --> $DIR/enum_variants.rs:14:5
   |
14 |     cFoo, //~ ERROR: Variant name ends with the enum's name
   |     ^^^^
   |
   = note: #[deny(enum_variant_names)] implied by #[deny(clippy)]
note: lint level defined here
  --> $DIR/enum_variants.rs:3:9
   |
3  | #![deny(clippy, pub_enum_variant_names)]
   |         ^^^^^^

error: Variant name starts with the enum's name
  --> $DIR/enum_variants.rs:25:5
   |
25 |     FoodGood, //~ ERROR: Variant name starts with the enum's name
   |     ^^^^^^^^
   |
   = note: #[deny(enum_variant_names)] implied by #[deny(clippy)]

error: Variant name starts with the enum's name
  --> $DIR/enum_variants.rs:26:5
   |
26 |     FoodMiddle, //~ ERROR: Variant name starts with the enum's name
   |     ^^^^^^^^^^
   |
   = note: #[deny(enum_variant_names)] implied by #[deny(clippy)]

error: Variant name starts with the enum's name
  --> $DIR/enum_variants.rs:27:5
   |
27 |     FoodBad, //~ ERROR: Variant name starts with the enum's name
   |     ^^^^^^^
   |
   = note: #[deny(enum_variant_names)] implied by #[deny(clippy)]

error: All variants have the same prefix: `Food`
  --> $DIR/enum_variants.rs:24:1
   |
24 |   enum Food { //~ ERROR: All variants have the same prefix: `Food`
   |  _^ starting here...
25 | |     FoodGood, //~ ERROR: Variant name starts with the enum's name
26 | |     FoodMiddle, //~ ERROR: Variant name starts with the enum's name
27 | |     FoodBad, //~ ERROR: Variant name starts with the enum's name
28 | | }
   | |_^ ...ending here
   |
   = note: #[deny(enum_variant_names)] implied by #[deny(clippy)]
   = help: remove the prefixes and use full paths to the variants instead of glob imports

error: All variants have the same prefix: `CallType`
  --> $DIR/enum_variants.rs:34:1
   |
34 |   enum BadCallType { //~ ERROR: All variants have the same prefix: `CallType`
   |  _^ starting here...
35 | |     CallTypeCall,
36 | |     CallTypeCreate,
37 | |     CallTypeDestroy,
38 | | }
   | |_^ ...ending here
   |
   = note: #[deny(enum_variant_names)] implied by #[deny(clippy)]
   = help: remove the prefixes and use full paths to the variants instead of glob imports

error: All variants have the same prefix: `Constant`
  --> $DIR/enum_variants.rs:45:1
   |
45 |   enum Consts { //~ ERROR: All variants have the same prefix: `Constant`
   |  _^ starting here...
46 | |     ConstantInt,
47 | |     ConstantCake,
48 | |     ConstantLie,
49 | | }
   | |_^ ...ending here
   |
   = note: #[deny(enum_variant_names)] implied by #[deny(clippy)]
   = help: remove the prefixes and use full paths to the variants instead of glob imports

error: All variants have the same prefix: `With`
  --> $DIR/enum_variants.rs:78:1
   |
78 |   enum Seallll { //~ ERROR: All variants have the same prefix: `With`
   |  _^ starting here...
79 | |     WithOutCake,
80 | |     WithOutTea,
81 | |     WithOut,
82 | | }
   | |_^ ...ending here
   |
   = note: #[deny(enum_variant_names)] implied by #[deny(clippy)]
   = help: remove the prefixes and use full paths to the variants instead of glob imports

error: All variants have the same prefix: `Prefix`
  --> $DIR/enum_variants.rs:84:1
   |
84 |   enum NonCaps { //~ ERROR: All variants have the same prefix: `Prefix`
   |  _^ starting here...
85 | |     Prefixçš„,
86 | |     PrefixTea,
87 | |     PrefixCake,
88 | | }
   | |_^ ...ending here
   |
   = note: #[deny(enum_variant_names)] implied by #[deny(clippy)]
   = help: remove the prefixes and use full paths to the variants instead of glob imports

error: All variants have the same prefix: `With`
  --> $DIR/enum_variants.rs:90:1
   |
90 |   pub enum PubSeall { //~ ERROR: All variants have the same prefix:
   |  _^ starting here...
91 | |     WithOutCake,
92 | |     WithOutTea,
93 | |     WithOut,
94 | | }
   | |_^ ...ending here
   |
note: lint level defined here
  --> $DIR/enum_variants.rs:3:17
   |
3  | #![deny(clippy, pub_enum_variant_names)]
   |                 ^^^^^^^^^^^^^^^^^^^^^^
   = help: remove the prefixes and use full paths to the variants instead of glob imports

error: aborting due to 10 previous errors

