error: local variable doesn't need to be boxed here
  --> $DIR/escape_analysis.rs:33:9
   |
33 |     let x = box A; //~ ERROR local variable
   |         ^
   |
note: lint level defined here
  --> $DIR/escape_analysis.rs:5:9
   |
5  | #![deny(boxed_local)]
   |         ^^^^^^^^^^^

error: local variable doesn't need to be boxed here
  --> $DIR/escape_analysis.rs:37:13
   |
37 | fn warn_arg(x: Box<A>) { //~ ERROR local variable
   |             ^

error: local variable doesn't need to be boxed here
  --> $DIR/escape_analysis.rs:49:9
   |
49 |     let y = x; //~ ERROR local variable
   |         ^

error: local variable doesn't need to be boxed here
  --> $DIR/escape_analysis.rs:54:9
   |
54 |     let bz = box A; //~ ERROR local variable
   |         ^^

error: local variable doesn't need to be boxed here
  --> $DIR/escape_analysis.rs:58:9
   |
58 |     let bz = box A; //~ ERROR local variable
   |         ^^

error: local variable doesn't need to be boxed here
  --> $DIR/escape_analysis.rs:88:9
   |
88 |     let x = box A; //~ ERROR local variable
   |         ^

error: local variable doesn't need to be boxed here
   --> $DIR/escape_analysis.rs:101:9
    |
101 |     let x = box A; //~ ERROR local variable
    |         ^

error: local variable doesn't need to be boxed here
   --> $DIR/escape_analysis.rs:130:12
    |
130 | pub fn new(_needs_name: Box<PeekableSeekable<&()>>) -> () { //~ ERROR local variable doesn't need
    |            ^^^^^^^^^^^

error: aborting due to 8 previous errors

