error: useless use of `format!`
 --> $DIR/format.rs:6:5
  |
6 |     format!("foo"); //~ERROR useless use of `format!`
  |     ^^^^^^^^^^^^^^^
  |
note: lint level defined here
 --> $DIR/format.rs:3:9
  |
3 | #![deny(useless_format)]
  |         ^^^^^^^^^^^^^^

error: useless use of `format!`
 --> $DIR/format.rs:8:5
  |
8 |     format!("{}", "foo"); //~ERROR useless use of `format!`
  |     ^^^^^^^^^^^^^^^^^^^^^

error: useless use of `format!`
  --> $DIR/format.rs:15:5
   |
15 |     format!("{}", arg); //~ERROR useless use of `format!`
   |     ^^^^^^^^^^^^^^^^^^^

error: aborting due to 3 previous errors

