error: the operation is ineffective. Consider reducing it to `x`
  --> $DIR/identity_op.rs:13:5
   |
13 |     x + 0;        //~ERROR the operation is ineffective
   |     ^^^^^
   |
note: lint level defined here
  --> $DIR/identity_op.rs:9:8
   |
9  | #[deny(identity_op)]
   |        ^^^^^^^^^^^

error: the operation is ineffective. Consider reducing it to `x`
  --> $DIR/identity_op.rs:14:5
   |
14 |     x + (1 - 1);  //~ERROR the operation is ineffective
   |     ^^^^^^^^^^^

error: the operation is ineffective. Consider reducing it to `x`
  --> $DIR/identity_op.rs:16:5
   |
16 |     0 + x;        //~ERROR the operation is ineffective
   |     ^^^^^

error: the operation is ineffective. Consider reducing it to `x`
  --> $DIR/identity_op.rs:19:5
   |
19 |     x | (0);      //~ERROR the operation is ineffective
   |     ^^^^^^^

error: the operation is ineffective. Consider reducing it to `x`
  --> $DIR/identity_op.rs:22:5
   |
22 |     x * 1;        //~ERROR the operation is ineffective
   |     ^^^^^

error: the operation is ineffective. Consider reducing it to `x`
  --> $DIR/identity_op.rs:23:5
   |
23 |     1 * x;        //~ERROR the operation is ineffective
   |     ^^^^^

error: the operation is ineffective. Consider reducing it to `x`
  --> $DIR/identity_op.rs:29:5
   |
29 |     -1 & x;       //~ERROR the operation is ineffective
   |     ^^^^^^

error: aborting due to 7 previous errors

