error: Unnecessary boolean `not` operation
  --> $DIR/if_not_else.rs:9:5
   |
9  |       if !bla() { //~ ERROR: Unnecessary boolean `not` operation
   |  _____^ starting here...
10 | |         println!("Bugs");
11 | |     } else {
12 | |         println!("Bunny");
13 | |     }
   | |_____^ ...ending here
   |
note: lint level defined here
  --> $DIR/if_not_else.rs:4:9
   |
4  | #![deny(if_not_else)]
   |         ^^^^^^^^^^^
   = help: remove the `!` and swap the blocks of the if/else

error: Unnecessary `!=` operation
  --> $DIR/if_not_else.rs:14:5
   |
14 |       if 4 != 5 { //~ ERROR: Unnecessary `!=` operation
   |  _____^ starting here...
15 | |         println!("Bugs");
16 | |     } else {
17 | |         println!("Bunny");
18 | |     }
   | |_____^ ...ending here
   |
   = help: change to `==` and swap the blocks of the if/else

error: aborting due to 2 previous errors

