error: adding items after statements is confusing, since items exist from the start of the scope
  --> $DIR/item_after_statement.rs:12:5
   |
12 |     fn foo() { println!("foo"); } //~ ERROR adding items after statements is confusing
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: lint level defined here
  --> $DIR/item_after_statement.rs:3:9
   |
3  | #![deny(items_after_statements)]
   |         ^^^^^^^^^^^^^^^^^^^^^^

error: adding items after statements is confusing, since items exist from the start of the scope
  --> $DIR/item_after_statement.rs:17:5
   |
17 |     fn foo() { println!("foo"); } //~ ERROR adding items after statements is confusing
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 2 previous errors

