error: `if _ { .. } else { .. }` is an expression
  --> $DIR/let_if_seq.rs:57:5
   |
57 |       let mut foo = 0;
   |  _____^ starting here...
58 | |     //~^ ERROR `if _ { .. } else { .. }` is an expression
59 | |     //~| HELP more idiomatic
60 | |     //~| SUGGESTION let <mut> foo = if f() { 42 } else { 0 };
61 | |     if f() {
62 | |         foo = 42;
63 | |     }
   | |_____^ ...ending here
   |
note: lint level defined here
  --> $DIR/let_if_seq.rs:5:9
   |
5  | #![deny(useless_let_if_seq)]
   |         ^^^^^^^^^^^^^^^^^^
help: it is more idiomatic to write
   |     let <mut> foo = if f() { 42 } else { 0 };
   = note: you might not need `mut` at all

error: `if _ { .. } else { .. }` is an expression
  --> $DIR/let_if_seq.rs:65:5
   |
65 |     let mut bar = 0;
   |     ^
   |
help: it is more idiomatic to write
   |     let <mut> bar = if f() { ..; 42 } else { ..; 0 };
   = note: you might not need `mut` at all

error: `if _ { .. } else { .. }` is an expression
  --> $DIR/let_if_seq.rs:77:5
   |
77 |     let quz;
   |     ^
   |
help: it is more idiomatic to write
   |     let quz = if f() { 42 } else { 0 };

error: `if _ { .. } else { .. }` is an expression
   --> $DIR/let_if_seq.rs:111:5
    |
111 |       let mut baz = 0;
    |  _____^ starting here...
112 | |     //~^ ERROR `if _ { .. } else { .. }` is an expression
113 | |     //~| HELP more idiomatic
114 | |     //~| SUGGESTION let <mut> baz = if f() { 42 } else { 0 };
115 | |     if f() {
116 | |         baz = 42;
117 | |     }
    | |_____^ ...ending here
    |
help: it is more idiomatic to write
    |     let <mut> baz = if f() { 42 } else { 0 };
    = note: you might not need `mut` at all

error: aborting due to 4 previous errors

