error: inconsistent casing in hexadecimal literal
  --> $DIR/literals.rs:14:17
   |
14 |     let fail1 = 0xabCD;       //~ERROR inconsistent casing in hexadecimal literal
   |                 ^^^^^^
   |
note: lint level defined here
  --> $DIR/literals.rs:3:9
   |
3  | #![deny(mixed_case_hex_literals)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^

error: inconsistent casing in hexadecimal literal
  --> $DIR/literals.rs:15:17
   |
15 |     let fail2 = 0xabCD_u32;   //~ERROR inconsistent casing in hexadecimal literal
   |                 ^^^^^^^^^^

error: inconsistent casing in hexadecimal literal
  --> $DIR/literals.rs:16:17
   |
16 |     let fail2 = 0xabCD_isize; //~ERROR inconsistent casing in hexadecimal literal
   |                 ^^^^^^^^^^^^

error: integer type suffix should be separated by an underscore
  --> $DIR/literals.rs:21:17
   |
21 |     let fail3 = 1234i32;      //~ERROR integer type suffix should be separated
   |                 ^^^^^^^
   |
note: lint level defined here
  --> $DIR/literals.rs:4:9
   |
4  | #![deny(unseparated_literal_suffix)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^

error: integer type suffix should be separated by an underscore
  --> $DIR/literals.rs:22:17
   |
22 |     let fail4 = 1234u32;      //~ERROR integer type suffix should be separated
   |                 ^^^^^^^

error: integer type suffix should be separated by an underscore
  --> $DIR/literals.rs:23:17
   |
23 |     let fail5 = 1234isize;    //~ERROR integer type suffix should be separated
   |                 ^^^^^^^^^

error: integer type suffix should be separated by an underscore
  --> $DIR/literals.rs:24:17
   |
24 |     let fail6 = 1234usize;    //~ERROR integer type suffix should be separated
   |                 ^^^^^^^^^

error: float type suffix should be separated by an underscore
  --> $DIR/literals.rs:25:17
   |
25 |     let fail7 = 1.5f32;       //~ERROR float type suffix should be separated
   |                 ^^^^^^

error: this is a decimal constant
  --> $DIR/literals.rs:29:17
   |
29 |     let fail8 = 0123;
   |                 ^^^^
   |
note: lint level defined here
  --> $DIR/literals.rs:5:9
   |
5  | #![deny(zero_prefixed_literal)]
   |         ^^^^^^^^^^^^^^^^^^^^^
help: if you mean to use a decimal constant, remove the `0` to remove confusion:
   |     let fail8 = 123;
help: if you mean to use an octal constant, use `0o`:
   |     let fail8 = 0o123;

error: aborting due to 9 previous errors

