error: The function/method "takes_an_immutable_reference" doesn't need a mutable reference
  --> $DIR/mut_reference.rs:22:34
   |
22 |     takes_an_immutable_reference(&mut 42); //~ERROR The function/method "takes_an_immutable_reference" doesn't need a mutable reference
   |                                  ^^^^^^^
   |
note: lint level defined here
  --> $DIR/mut_reference.rs:19:8
   |
19 | #[deny(unnecessary_mut_passed)]
   |        ^^^^^^^^^^^^^^^^^^^^^^

error: The function/method "as_ptr" doesn't need a mutable reference
  --> $DIR/mut_reference.rs:24:12
   |
24 |     as_ptr(&mut 42); //~ERROR The function/method "as_ptr" doesn't need a mutable reference
   |            ^^^^^^^

error: The function/method "takes_an_immutable_reference" doesn't need a mutable reference
  --> $DIR/mut_reference.rs:28:44
   |
28 |     my_struct.takes_an_immutable_reference(&mut 42); //~ERROR The function/method "takes_an_immutable_reference" doesn't need a mutable reference
   |                                            ^^^^^^^

error: aborting due to 3 previous errors

