error: you probably are missing some parameter in your format string
 --> $DIR/panic.rs:8:16
  |
8 |         panic!("{}"); //~ERROR: you probably are missing some parameter
  |                ^^^^
  |
note: lint level defined here
 --> $DIR/panic.rs:4:9
  |
4 | #![deny(panic_params)]
  |         ^^^^^^^^^^^^

error: you probably are missing some parameter in your format string
  --> $DIR/panic.rs:10:16
   |
10 |         panic!("{:?}"); //~ERROR: you probably are missing some parameter
   |                ^^^^^^

error: you probably are missing some parameter in your format string
  --> $DIR/panic.rs:12:23
   |
12 |         assert!(true, "here be missing values: {}"); //~ERROR you probably are missing some parameter
   |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 3 previous errors

