error: using `print!()` with a format string that ends in a newline, consider using `println!()` instead
 --> $DIR/print_with_newline.rs:6:5
  |
6 |     print!("Hello/n"); //~ERROR using `print!()` with a format string
  |     ^^^^^^^^^^^^^^^^^^
  |
note: lint level defined here
 --> $DIR/print_with_newline.rs:3:9
  |
3 | #![deny(print_with_newline)]
  |         ^^^^^^^^^^^^^^^^^^

error: using `print!()` with a format string that ends in a newline, consider using `println!()` instead
 --> $DIR/print_with_newline.rs:7:5
  |
7 |     print!("Hello {}/n", "world"); //~ERROR using `print!()` with a format string
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: using `print!()` with a format string that ends in a newline, consider using `println!()` instead
 --> $DIR/print_with_newline.rs:8:5
  |
8 |     print!("Hello {} {}/n/n", "world", "#2"); //~ERROR using `print!()` with a format string
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: using `print!()` with a format string that ends in a newline, consider using `println!()` instead
 --> $DIR/print_with_newline.rs:9:5
  |
9 |     print!("{}/n", 1265); //~ERROR using `print!()` with a format string
  |     ^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 4 previous errors

