error: item name starts with its containing module's name
 --> $DIR/stutter.rs:8:5
  |
8 |     pub fn foo_bar() {} //~ ERROR: item name starts with its containing module's name
  |     ^^^^^^^^^^^^^^^^^^^
  |
note: lint level defined here
 --> $DIR/stutter.rs:3:9
  |
3 | #![deny(stutter)]
  |         ^^^^^^^

error: item name ends with its containing module's name
 --> $DIR/stutter.rs:9:5
  |
9 |     pub fn bar_foo() {} //~ ERROR: item name ends with its containing module's name
  |     ^^^^^^^^^^^^^^^^^^^

error: item name starts with its containing module's name
  --> $DIR/stutter.rs:10:5
   |
10 |     pub struct FooCake {} //~ ERROR: item name starts with its containing module's name
   |     ^^^^^^^^^^^^^^^^^^^^^

error: item name ends with its containing module's name
  --> $DIR/stutter.rs:11:5
   |
11 |     pub enum CakeFoo {} //~ ERROR: item name ends with its containing module's name
   |     ^^^^^^^^^^^^^^^^^^^

error: aborting due to 4 previous errors

