error: assignment to temporary
  --> $DIR/temporary_assignment.rs:29:5
   |
29 |     Struct { field: 0 }.field = 1; //~ERROR assignment to temporary
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: lint level defined here
  --> $DIR/temporary_assignment.rs:4:9
   |
4  | #![deny(temporary_assignment)]
   |         ^^^^^^^^^^^^^^^^^^^^

error: assignment to temporary
  --> $DIR/temporary_assignment.rs:30:5
   |
30 |     (0, 0).0 = 1; //~ERROR assignment to temporary
   |     ^^^^^^^^^^^^

error: aborting due to 2 previous errors

