error: ==-comparison of unit values detected. This will always be true
  --> $DIR/unit_cmp.rs:16:8
   |
16 |     if { true; } == { false; } {  //~ERROR ==-comparison of unit values detected. This will always be true
   |        ^^^^^^^^^^^^^^^^^^^^^^^
   |
note: lint level defined here
  --> $DIR/unit_cmp.rs:4:9
   |
4  | #![deny(unit_cmp)]
   |         ^^^^^^^^

error: >-comparison of unit values detected. This will always be false
  --> $DIR/unit_cmp.rs:19:8
   |
19 |     if { true; } > { false; } {  //~ERROR >-comparison of unit values detected. This will always be false
   |        ^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 2 previous errors

