error: this lifetime isn't used in the function definition
  --> $DIR/unused_lt.rs:16:14
   |
16 | fn unused_lt<'a>(x: u8) { //~ ERROR this lifetime
   |              ^^
   |
note: lint level defined here
  --> $DIR/unused_lt.rs:4:9
   |
4  | #![deny(unused_lifetimes)]
   |         ^^^^^^^^^^^^^^^^

error: this lifetime isn't used in the function definition
  --> $DIR/unused_lt.rs:20:25
   |
20 | fn unused_lt_transitive<'a, 'b: 'a>(x: &'b u8) { //~ ERROR this lifetime
   |                         ^^

error: this lifetime isn't used in the function definition
  --> $DIR/unused_lt.rs:50:10
   |
50 |     fn x<'a>(&self) {} //~ ERROR this lifetime
   |          ^^

error: aborting due to 3 previous errors

