error[E0252]: the name `foo` is defined multiple times
  --> $DIR/duplicate.rs:25:9
   |
24 |     use a::foo; //~ NOTE previous import of the value `foo` here
   |         ------ previous import of the value `foo` here
25 |     use a::foo; //~ ERROR the name `foo` is defined multiple times
   |         ^^^^^^ `foo` reimported here
   |
   = note: `foo` must be defined only once in the value namespace of this module
help: You can use `as` to change the binding name of the import
   |
25 |     use a::foo as Otherfoo; //~ ERROR the name `foo` is defined multiple times
   |         ^^^^^^^^^^^^^^^^^^

error: `foo` is ambiguous
  --> $DIR/duplicate.rs:62:9
   |
62 |     use self::foo::bar; //~ ERROR `foo` is ambiguous
   |         ^^^^^^^^^^^^^^
   |
note: `foo` could refer to the name imported here
  --> $DIR/duplicate.rs:57:9
   |
57 |     use self::m1::*; //~ NOTE
   |         ^^^^^^^^^^^
note: `foo` could also refer to the name imported here
  --> $DIR/duplicate.rs:59:9
   |
59 |     use self::m2::*; //~ NOTE
   |         ^^^^^^^^^^^
   = note: consider adding an explicit import of `foo` to disambiguate

error: `foo` is ambiguous
  --> $DIR/duplicate.rs:47:5
   |
47 |     f::foo(); //~ ERROR `foo` is ambiguous
   |     ^^^^^^
   |
note: `foo` could refer to the name imported here
  --> $DIR/duplicate.rs:36:13
   |
36 |     pub use a::*; //~ NOTE `foo` could refer to the name imported here
   |             ^^^^
note: `foo` could also refer to the name imported here
  --> $DIR/duplicate.rs:37:13
   |
37 |     pub use b::*; //~ NOTE `foo` could also refer to the name imported here
   |             ^^^^
   = note: consider adding an explicit import of `foo` to disambiguate

error: `foo` is ambiguous
  --> $DIR/duplicate.rs:49:5
   |
49 |     g::foo(); //~ ERROR `foo` is ambiguous
   |     ^^^^^^
   |
note: `foo` could refer to the name imported here
  --> $DIR/duplicate.rs:41:13
   |
41 |     pub use a::*; //~ NOTE `foo` could refer to the name imported here
   |             ^^^^
note: `foo` could also refer to the name imported here
  --> $DIR/duplicate.rs:42:13
   |
42 |     pub use f::*; //~ NOTE `foo` could also refer to the name imported here
   |             ^^^^
   = note: consider adding an explicit import of `foo` to disambiguate

error: `foo` is ambiguous
  --> $DIR/duplicate.rs:66:9
   |
66 |         foo::bar(); //~ ERROR `foo` is ambiguous
   |         ^^^^^^^^
   |
note: `foo` could refer to the name imported here
  --> $DIR/duplicate.rs:57:9
   |
57 |     use self::m1::*; //~ NOTE
   |         ^^^^^^^^^^^
note: `foo` could also refer to the name imported here
  --> $DIR/duplicate.rs:59:9
   |
59 |     use self::m2::*; //~ NOTE
   |         ^^^^^^^^^^^
   = note: consider adding an explicit import of `foo` to disambiguate

error: aborting due to 5 previous errors

