error: `bar` is ambiguous
  --> $DIR/macro-paths.rs:25:5
   |
25 |     bar::m! { //~ ERROR ambiguous
   |     ^^^^^^
   |
note: `bar` could refer to the name defined here
  --> $DIR/macro-paths.rs:27:9
   |
27 |         mod bar { pub use two_macros::m; } //~ NOTE could refer to the name defined here
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
note: `bar` could also refer to the name imported here
  --> $DIR/macro-paths.rs:24:9
   |
24 |     use foo::*; //~ NOTE could also refer to the name imported here
   |         ^^^^^^
   = note: macro-expanded items do not shadow when used in a macro invocation path

error: `baz` is ambiguous
  --> $DIR/macro-paths.rs:36:5
   |
36 |     baz::m! { //~ ERROR ambiguous
   |     ^^^^^^
   |
note: `baz` could refer to the name defined here
  --> $DIR/macro-paths.rs:38:9
   |
38 |         mod baz { pub use two_macros::m; } //~ NOTE could refer to the name defined here
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
note: `baz` could also refer to the name defined here
  --> $DIR/macro-paths.rs:31:1
   |
31 | / pub mod baz { //~ NOTE could also refer to the name defined here
32 | |     pub use two_macros::m;
33 | | }
   | |_^
   = note: macro-expanded items do not shadow when used in a macro invocation path

error[E0601]: main function not found

error: aborting due to 3 previous errors

