error: any use of this value will cause an error
  --> $DIR/const-eval-overflow2.rs:24:1
   |
LL | / const VALS_I8: (i8,) = //~ ERROR any use of this value will cause an error
LL | |     (
LL | |      i8::MIN - 1,
   | |      ----------- attempt to subtract with overflow
LL | |      );
   | |_______^
   |
note: lint level defined here
  --> $DIR/const-eval-overflow2.rs:18:9
   |
LL | #![deny(const_err)]
   |         ^^^^^^^^^

error: any use of this value will cause an error
  --> $DIR/const-eval-overflow2.rs:29:1
   |
LL | / const VALS_I16: (i16,) = //~ ERROR any use of this value will cause an error
LL | |     (
LL | |      i16::MIN - 1,
   | |      ------------ attempt to subtract with overflow
LL | |      );
   | |_______^

error: any use of this value will cause an error
  --> $DIR/const-eval-overflow2.rs:34:1
   |
LL | / const VALS_I32: (i32,) = //~ ERROR any use of this value will cause an error
LL | |     (
LL | |      i32::MIN - 1,
   | |      ------------ attempt to subtract with overflow
LL | |      );
   | |_______^

error: any use of this value will cause an error
  --> $DIR/const-eval-overflow2.rs:39:1
   |
LL | / const VALS_I64: (i64,) = //~ ERROR any use of this value will cause an error
LL | |     (
LL | |      i64::MIN - 1,
   | |      ------------ attempt to subtract with overflow
LL | |      );
   | |_______^

error: any use of this value will cause an error
  --> $DIR/const-eval-overflow2.rs:44:1
   |
LL | / const VALS_U8: (u8,) = //~ ERROR any use of this value will cause an error
LL | |     (
LL | |      u8::MIN - 1,
   | |      ----------- attempt to subtract with overflow
LL | |      );
   | |_______^

error: any use of this value will cause an error
  --> $DIR/const-eval-overflow2.rs:49:1
   |
LL | / const VALS_U16: (u16,) = ( //~ ERROR any use of this value will cause an error
LL | |      u16::MIN - 1,
   | |      ------------ attempt to subtract with overflow
LL | |      );
   | |_______^

error: any use of this value will cause an error
  --> $DIR/const-eval-overflow2.rs:53:1
   |
LL | / const VALS_U32: (u32,) = ( //~ ERROR any use of this value will cause an error
LL | |      u32::MIN - 1,
   | |      ------------ attempt to subtract with overflow
LL | |      );
   | |_______^

error: any use of this value will cause an error
  --> $DIR/const-eval-overflow2.rs:57:1
   |
LL | / const VALS_U64: (u64,) = //~ ERROR any use of this value will cause an error
LL | |     (
LL | |      u64::MIN - 1,
   | |      ------------ attempt to subtract with overflow
LL | |      );
   | |_______^

error[E0080]: erroneous constant used
  --> $DIR/const-eval-overflow2.rs:63:5
   |
LL |     foo(VALS_I8); //~ ERROR erroneous constant used
   |     ^^^^^^^^^^^^ referenced constant has errors

error[E0080]: erroneous constant used
  --> $DIR/const-eval-overflow2.rs:64:5
   |
LL |     foo(VALS_I16); //~ ERROR erroneous constant used
   |     ^^^^^^^^^^^^^ referenced constant has errors

error[E0080]: erroneous constant used
  --> $DIR/const-eval-overflow2.rs:65:5
   |
LL |     foo(VALS_I32); //~ ERROR erroneous constant used
   |     ^^^^^^^^^^^^^ referenced constant has errors

error[E0080]: erroneous constant used
  --> $DIR/const-eval-overflow2.rs:66:5
   |
LL |     foo(VALS_I64); //~ ERROR erroneous constant used
   |     ^^^^^^^^^^^^^ referenced constant has errors

error[E0080]: erroneous constant used
  --> $DIR/const-eval-overflow2.rs:68:5
   |
LL |     foo(VALS_U8); //~ ERROR erroneous constant used
   |     ^^^^^^^^^^^^ referenced constant has errors

error[E0080]: erroneous constant used
  --> $DIR/const-eval-overflow2.rs:69:5
   |
LL |     foo(VALS_U16); //~ ERROR erroneous constant used
   |     ^^^^^^^^^^^^^ referenced constant has errors

error[E0080]: erroneous constant used
  --> $DIR/const-eval-overflow2.rs:70:5
   |
LL |     foo(VALS_U32); //~ ERROR erroneous constant used
   |     ^^^^^^^^^^^^^ referenced constant has errors

error[E0080]: erroneous constant used
  --> $DIR/const-eval-overflow2.rs:71:5
   |
LL |     foo(VALS_U64); //~ ERROR erroneous constant used
   |     ^^^^^^^^^^^^^ referenced constant has errors

error: aborting due to 16 previous errors

For more information about this error, try `rustc --explain E0080`.
