error[E0716]: temporary value dropped while borrowed
  --> $DIR/promoted_raw_ptr_ops.rs:14:29
   |
LL |     let x: &'static bool = &(42 as *const i32 == 43 as *const i32);
   |            -------------    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ creates a temporary which is freed while still in use
   |            |
   |            type annotation requires that borrow lasts for `'static`
...
LL | }
   | - temporary value is freed at the end of this statement

error[E0716]: temporary value dropped while borrowed
  --> $DIR/promoted_raw_ptr_ops.rs:16:30
   |
LL |     let y: &'static usize = &(&1 as *const i32 as usize + 1); //~ ERROR does not live long enough
   |            --------------    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ creates a temporary which is freed while still in use
   |            |
   |            type annotation requires that borrow lasts for `'static`
...
LL | }
   | - temporary value is freed at the end of this statement

error[E0716]: temporary value dropped while borrowed
  --> $DIR/promoted_raw_ptr_ops.rs:18:29
   |
LL |     let a: &'static bool = &(main as fn() == main as fn()); //~ ERROR does not live long enough
   |            -------------    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ creates a temporary which is freed while still in use
   |            |
   |            type annotation requires that borrow lasts for `'static`
LL | }
   | - temporary value is freed at the end of this statement

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0716`.
