error: union expressions should have exactly one field
  --> $DIR/union-fields-2.rs:17:13
   |
17 |     let u = U {}; //~ ERROR union expressions should have exactly one field
   |             ^

error: union expressions should have exactly one field
  --> $DIR/union-fields-2.rs:19:13
   |
19 |     let u = U { a: 0, b: 1 }; //~ ERROR union expressions should have exactly one field
   |             ^

error[E0560]: union `U` has no field named `c`
  --> $DIR/union-fields-2.rs:20:29
   |
20 |     let u = U { a: 0, b: 1, c: 2 }; //~ ERROR union expressions should have exactly one field
   |                             ^^ `U` does not have this field
   |
   = note: available fields are: `a`, `b`

error: union expressions should have exactly one field
  --> $DIR/union-fields-2.rs:20:13
   |
20 |     let u = U { a: 0, b: 1, c: 2 }; //~ ERROR union expressions should have exactly one field
   |             ^

error: union expressions should have exactly one field
  --> $DIR/union-fields-2.rs:24:13
   |
24 |     let u = U { ..u }; //~ ERROR union expressions should have exactly one field
   |             ^

error[E0436]: functional record update syntax requires a struct
  --> $DIR/union-fields-2.rs:24:19
   |
24 |     let u = U { ..u }; //~ ERROR union expressions should have exactly one field
   |                   ^

error: union patterns should have exactly one field
  --> $DIR/union-fields-2.rs:27:9
   |
27 |     let U {} = u; //~ ERROR union patterns should have exactly one field
   |         ^^^^

error: union patterns should have exactly one field
  --> $DIR/union-fields-2.rs:29:9
   |
29 |     let U { a, b } = u; //~ ERROR union patterns should have exactly one field
   |         ^^^^^^^^^^

error[E0026]: union `U` does not have a field named `c`
  --> $DIR/union-fields-2.rs:30:19
   |
30 |     let U { a, b, c } = u; //~ ERROR union patterns should have exactly one field
   |                   ^ union `U` does not have field `c`

error: union patterns should have exactly one field
  --> $DIR/union-fields-2.rs:30:9
   |
30 |     let U { a, b, c } = u; //~ ERROR union patterns should have exactly one field
   |         ^^^^^^^^^^^^^

error: union patterns should have exactly one field
  --> $DIR/union-fields-2.rs:33:9
   |
33 |     let U { .. } = u; //~ ERROR union patterns should have exactly one field
   |         ^^^^^^^^

error: `..` cannot be used in union patterns
  --> $DIR/union-fields-2.rs:33:9
   |
33 |     let U { .. } = u; //~ ERROR union patterns should have exactly one field
   |         ^^^^^^^^

error: `..` cannot be used in union patterns
  --> $DIR/union-fields-2.rs:35:9
   |
35 |     let U { a, .. } = u; //~ ERROR `..` cannot be used in union patterns
   |         ^^^^^^^^^^^

error: aborting due to 13 previous errors

