warning: the feature `const_generics` is incomplete and may cause the compiler to crash
  --> $DIR/const-param-type-depends-on-type-param.rs:1:12
   |
LL | #![feature(const_generics)]
   |            ^^^^^^^^^^^^^^

error[E0670]: const parameters cannot depend on type parameters
  --> $DIR/const-param-type-depends-on-type-param.rs:4:34
   |
LL | pub struct Dependent<T, const X: T>([(); X]); //~ ERROR const parameters
   |                                  ^ const parameter depends on type parameter

error: const generics in any position are currently unsupported
  --> $DIR/const-param-type-depends-on-type-param.rs:4:31
   |
LL | pub struct Dependent<T, const X: T>([(); X]); //~ ERROR const parameters
   |                               ^

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0670`.
