error[E0423]: expected value, found struct `Z`
  --> $DIR/privacy-struct-ctor.rs:29:9
   |
29 |         Z;
   |         ^
   |         |
   |         did you mean `S`?
   |         constructor is not visible here due to private fields
help: possible better candidate is found in another module, you can import it into scope
   |
25 |     use m::n::Z;
   |

error[E0423]: expected value, found struct `S`
  --> $DIR/privacy-struct-ctor.rs:39:5
   |
39 |     S;
   |     ^ constructor is not visible here due to private fields
help: possible better candidate is found in another module, you can import it into scope
   |
34 | use m::S;
   |

error[E0423]: expected value, found struct `S2`
  --> $DIR/privacy-struct-ctor.rs:43:5
   |
43 |     S2;
   |     ^^ did you mean `S2 { /* fields */ }`?

error[E0423]: expected value, found struct `xcrate::S`
  --> $DIR/privacy-struct-ctor.rs:47:5
   |
47 |     xcrate::S;
   |     ^^^^^^^^^ constructor is not visible here due to private fields
help: possible better candidate is found in another module, you can import it into scope
   |
34 | use m::S;
   |

error[E0603]: tuple struct `Z` is private
  --> $DIR/privacy-struct-ctor.rs:28:9
   |
28 |         n::Z; //~ ERROR tuple struct `Z` is private
   |         ^^^^

error[E0603]: tuple struct `S` is private
  --> $DIR/privacy-struct-ctor.rs:38:5
   |
38 |     m::S; //~ ERROR tuple struct `S` is private
   |     ^^^^

error[E0603]: tuple struct `Z` is private
  --> $DIR/privacy-struct-ctor.rs:41:5
   |
41 |     m::n::Z; //~ ERROR tuple struct `Z` is private
   |     ^^^^^^^

error[E0603]: tuple struct `S` is private
  --> $DIR/privacy-struct-ctor.rs:46:5
   |
46 |     xcrate::m::S; //~ ERROR tuple struct `S` is private
   |     ^^^^^^^^^^^^

error[E0603]: tuple struct `Z` is private
  --> $DIR/privacy-struct-ctor.rs:49:5
   |
49 |     xcrate::m::n::Z; //~ ERROR tuple struct `Z` is private
   |     ^^^^^^^^^^^^^^^

error: aborting due to 9 previous errors

