warning: not reporting region error due to nll
  --> $DIR/regions-return-ref-to-upvar-issue-17403.rs:17:24
   |
LL |         let mut f = || &mut x; //~ ERROR cannot infer
   |                        ^^^^^^

error: unsatisfied lifetime constraints
  --> $DIR/regions-return-ref-to-upvar-issue-17403.rs:17:24
   |
LL |         let mut f = || &mut x; //~ ERROR cannot infer
   |                     ---^^^^^^
   |                     |  |
   |                     |  return requires that `'1` must outlive `'2`
   |                     lifetime `'1` represents the closure body
   |                     lifetime `'2` appears in return type

error: aborting due to previous error

