error[E0603]: static `private` is private
  --> $DIR/static-priv-by-default.rs:25:9
   |
LL |     foo(static_priv_by_default::private); //~ ERROR: static `private` is private
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0603]: static `private` is private
  --> $DIR/static-priv-by-default.rs:27:9
   |
LL |     foo(child::childs_child::private); //~ ERROR: static `private` is private
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0603]: static `private` is private
  --> $DIR/static-priv-by-default.rs:33:9
   |
LL |     foo(childs_child::private); //~ ERROR: static `private` is private
   |         ^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0603`.
