error[E0658]: The attribute `C` is currently unknown to the compiler and may have meaning added to it in the future (see issue #29642)
  --> $DIR/proc-macro-attributes.rs:7:3
   |
LL | #[C] //~ ERROR attribute `C` is currently unknown to the compiler
   |   ^
   |
   = help: add #![feature(custom_attribute)] to the crate attributes to enable

error[E0659]: `B` is ambiguous (derive helper attribute vs any other name)
  --> $DIR/proc-macro-attributes.rs:16:3
   |
LL | #[B] //~ ERROR `B` is ambiguous
   |   ^ ambiguous name
   |
note: `B` could refer to the derive helper attribute defined here
  --> $DIR/proc-macro-attributes.rs:22:10
   |
LL | #[derive(B)]
   |          ^
note: `B` could also refer to the derive macro imported here
  --> $DIR/proc-macro-attributes.rs:13:1
   |
LL | #[macro_use]
   | ^^^^^^^^^^^^

error[E0659]: `B` is ambiguous (derive helper attribute vs any other name)
  --> $DIR/proc-macro-attributes.rs:18:3
   |
LL | #[B(D)] //~ ERROR `B` is ambiguous
   |   ^ ambiguous name
   |
note: `B` could refer to the derive helper attribute defined here
  --> $DIR/proc-macro-attributes.rs:22:10
   |
LL | #[derive(B)]
   |          ^
note: `B` could also refer to the derive macro imported here
  --> $DIR/proc-macro-attributes.rs:13:1
   |
LL | #[macro_use]
   | ^^^^^^^^^^^^

error[E0659]: `B` is ambiguous (derive helper attribute vs any other name)
  --> $DIR/proc-macro-attributes.rs:19:3
   |
LL | #[B(E = "foo")] //~ ERROR `B` is ambiguous
   |   ^ ambiguous name
   |
note: `B` could refer to the derive helper attribute defined here
  --> $DIR/proc-macro-attributes.rs:22:10
   |
LL | #[derive(B)]
   |          ^
note: `B` could also refer to the derive macro imported here
  --> $DIR/proc-macro-attributes.rs:13:1
   |
LL | #[macro_use]
   | ^^^^^^^^^^^^

error[E0659]: `B` is ambiguous (derive helper attribute vs any other name)
  --> $DIR/proc-macro-attributes.rs:20:3
   |
LL | #[B(arbitrary tokens)] //~ ERROR `B` is ambiguous
   |   ^ ambiguous name
   |
note: `B` could refer to the derive helper attribute defined here
  --> $DIR/proc-macro-attributes.rs:22:10
   |
LL | #[derive(B)]
   |          ^
note: `B` could also refer to the derive macro imported here
  --> $DIR/proc-macro-attributes.rs:13:1
   |
LL | #[macro_use]
   | ^^^^^^^^^^^^

error: expected one of `(`, `)`, `,`, `::`, or `=`, found `tokens`
  --> $DIR/proc-macro-attributes.rs:20:15
   |
LL | #[B(arbitrary tokens)] //~ ERROR `B` is ambiguous
   |               ^^^^^^ expected one of `(`, `)`, `,`, `::`, or `=` here

error: aborting due to 6 previous errors

Some errors occurred: E0658, E0659.
For more information about an error, try `rustc --explain E0658`.
