error: mutable references in const fn are unstable
  --> $DIR/mutable_borrow.rs:3:9
   |
LL |     let b = &mut a; //~ ERROR mutable references in const fn are unstable
   |         ^

error: mutable references in const fn are unstable
  --> $DIR/mutable_borrow.rs:12:13
   |
LL |         let b = &mut a; //~ ERROR mutable references in const fn are unstable
   |             ^

error[E0017]: references in statics may only refer to immutable values
  --> $DIR/mutable_borrow.rs:19:13
   |
LL |     let b = &mut a; //~ references in statics may only refer to immutable
   |             ^^^^^^ statics require immutable values

error[E0017]: references in statics may only refer to immutable values
  --> $DIR/mutable_borrow.rs:26:15
   |
LL |     { let b = &mut a; }  //~ references in statics may only refer to immutable
   |               ^^^^^^ statics require immutable values

error[E0017]: references in statics may only refer to immutable values
  --> $DIR/mutable_borrow.rs:37:17
   |
LL |     let mut a = &mut None; //~ references in statics may only refer to immutable values
   |                 ^^^^^^^^^ statics require immutable values

error[E0019]: static contains unimplemented expression type
  --> $DIR/mutable_borrow.rs:39:5
   |
LL |     *a = Some(Foo); //~ unimplemented expression type
   |     ^^

error[E0716]: temporary value dropped while borrowed
  --> $DIR/mutable_borrow.rs:37:22
   |
LL |     let mut a = &mut None; //~ references in statics may only refer to immutable values
   |                      ^^^^ creates a temporary which is freed while still in use
...
LL |     a
   |     - using this value as a static requires that borrow lasts for `'static`
LL | };
   | - temporary value is freed at the end of this statement

error: aborting due to 7 previous errors

Some errors occurred: E0017, E0019, E0716.
For more information about an error, try `rustc --explain E0017`.
