error[E0080]: constant evaluation error
  --> $DIR/issue-43197.rs:20:23
   |
LL |     println!("{} {}", X, Y);
   |                       ^ referenced constant has errors

error[E0080]: constant evaluation error
  --> $DIR/issue-43197.rs:20:26
   |
LL |     println!("{} {}", X, Y);
   |                          ^ referenced constant has errors

error[E0080]: constant evaluation error
  --> $DIR/issue-43197.rs:19:24
   |
LL |     const Y: u32 = foo(0-1); //~ ERROR constant evaluation error
   |                        ^^^ attempt to subtract with overflow

error[E0080]: constant evaluation error
  --> $DIR/issue-43197.rs:18:20
   |
LL |     const X: u32 = 0-1; //~ ERROR constant evaluation error
   |                    ^^^ attempt to subtract with overflow

error: aborting due to 4 previous errors

If you want more information on this error, try using "rustc --explain E0080"
