error[E0271]: type mismatch resolving `for<'x> <UintStruct as TheTrait<&'x isize>>::A == &'x isize`
  --> $DIR/associated-types-eq-hr.rs:82:5
   |
LL |     foo::<UintStruct>(); //~ ERROR type mismatch
   |     ^^^^^^^^^^^^^^^^^ expected usize, found isize
   |
   = note: expected type `&usize`
              found type `&isize`
note: required by `foo`
  --> $DIR/associated-types-eq-hr.rs:44:1
   |
LL | / fn foo<T>()
LL | |     where T : for<'x> TheTrait<&'x isize, A = &'x isize>
LL | | {
LL | |     // ok for IntStruct, but not UintStruct
LL | | }
   | |_^

error[E0271]: type mismatch resolving `for<'x> <IntStruct as TheTrait<&'x isize>>::A == &'x usize`
  --> $DIR/associated-types-eq-hr.rs:86:5
   |
LL |     bar::<IntStruct>(); //~ ERROR type mismatch
   |     ^^^^^^^^^^^^^^^^ expected isize, found usize
   |
   = note: expected type `&isize`
              found type `&usize`
note: required by `bar`
  --> $DIR/associated-types-eq-hr.rs:50:1
   |
LL | / fn bar<T>()
LL | |     where T : for<'x> TheTrait<&'x isize, A = &'x usize>
LL | | {
LL | |     // ok for UintStruct, but not IntStruct
LL | | }
   | |_^

error[E0495]: cannot infer an appropriate lifetime for lifetime parameter `'a` due to conflicting requirements
  --> $DIR/associated-types-eq-hr.rs:91:5
   |
LL |     tuple_one::<Tuple>();
   |     ^^^^^^^^^^^^^^^^^^
   |
   = note: first, the lifetime cannot outlive lifetime RePlaceholder(Placeholder { universe: U1, name: BrNamed(crate0:DefIndex(1:20), 'x) })...
   = note: ...but the lifetime must also be valid for lifetime RePlaceholder(Placeholder { universe: U1, name: BrNamed(crate0:DefIndex(1:21), 'y) })...
   = note: ...so that the types are compatible:
           expected TheTrait<(&'x isize, &'y isize)>
              found TheTrait<(&isize, &isize)>

error[E0495]: cannot infer an appropriate lifetime for lifetime parameter `'a` due to conflicting requirements
  --> $DIR/associated-types-eq-hr.rs:91:5
   |
LL |     tuple_one::<Tuple>();
   |     ^^^^^^^^^^^^^^^^^^
   |
   = note: first, the lifetime cannot outlive lifetime RePlaceholder(Placeholder { universe: U2, name: BrNamed(crate0:DefIndex(1:20), 'x) })...
   = note: ...but the lifetime must also be valid for lifetime RePlaceholder(Placeholder { universe: U2, name: BrNamed(crate0:DefIndex(1:21), 'y) })...
   = note: ...so that the types are compatible:
           expected TheTrait<(&'x isize, &'y isize)>
              found TheTrait<(&isize, &isize)>

error[E0495]: cannot infer an appropriate lifetime for lifetime parameter `'a` due to conflicting requirements
  --> $DIR/associated-types-eq-hr.rs:97:5
   |
LL |     tuple_two::<Tuple>();
   |     ^^^^^^^^^^^^^^^^^^
   |
   = note: first, the lifetime cannot outlive lifetime RePlaceholder(Placeholder { universe: U1, name: BrNamed(crate0:DefIndex(1:23), 'x) })...
   = note: ...but the lifetime must also be valid for lifetime RePlaceholder(Placeholder { universe: U1, name: BrNamed(crate0:DefIndex(1:24), 'y) })...
   = note: ...so that the types are compatible:
           expected TheTrait<(&'x isize, &'y isize)>
              found TheTrait<(&isize, &isize)>

error[E0495]: cannot infer an appropriate lifetime for lifetime parameter `'a` due to conflicting requirements
  --> $DIR/associated-types-eq-hr.rs:97:5
   |
LL |     tuple_two::<Tuple>();
   |     ^^^^^^^^^^^^^^^^^^
   |
   = note: first, the lifetime cannot outlive lifetime RePlaceholder(Placeholder { universe: U2, name: BrNamed(crate0:DefIndex(1:23), 'x) })...
   = note: ...but the lifetime must also be valid for lifetime RePlaceholder(Placeholder { universe: U2, name: BrNamed(crate0:DefIndex(1:24), 'y) })...
   = note: ...so that the types are compatible:
           expected TheTrait<(&'x isize, &'y isize)>
              found TheTrait<(&isize, &isize)>

error[E0495]: cannot infer an appropriate lifetime for lifetime parameter `'a` due to conflicting requirements
  --> $DIR/associated-types-eq-hr.rs:107:5
   |
LL |     tuple_four::<Tuple>();
   |     ^^^^^^^^^^^^^^^^^^^
   |
   = note: first, the lifetime cannot outlive lifetime RePlaceholder(Placeholder { universe: U1, name: BrNamed(crate0:DefIndex(1:28), 'x) })...
   = note: ...but the lifetime must also be valid for lifetime RePlaceholder(Placeholder { universe: U1, name: BrNamed(crate0:DefIndex(1:29), 'y) })...
   = note: ...so that the types are compatible:
           expected TheTrait<(&'x isize, &'y isize)>
              found TheTrait<(&isize, &isize)>

error: aborting due to 7 previous errors

Some errors occurred: E0271, E0495.
For more information about an error, try `rustc --explain E0271`.
