error[E0623]: lifetime mismatch
  --> $DIR/regions-lifetime-bounds-on-fns.rs:8:10
   |
LL | fn b<'a, 'b>(x: &mut &'a isize, y: &mut &'b isize) {
   |                      ---------          --------- these two types are declared with different lifetimes...
LL |     // Illegal now because there is no `'b:'a` declaration.
LL |     *x = *y; //~ ERROR E0623
   |          ^^ ...but data from `y` flows into `x` here

error[E0623]: lifetime mismatch
  --> $DIR/regions-lifetime-bounds-on-fns.rs:14:7
   |
LL | fn c<'a,'b>(x: &mut &'a isize, y: &mut &'b isize) {
   |                     ---------          --------- these two types are declared with different lifetimes...
...
LL |     a(x, y); //~ ERROR lifetime mismatch [E0623]
   |       ^ ...but data from `y` flows into `x` here

error[E0495]: cannot infer an appropriate lifetime for lifetime parameter `'a` due to conflicting requirements
  --> $DIR/regions-lifetime-bounds-on-fns.rs:20:43
   |
LL |     let _: fn(&mut &isize, &mut &isize) = a; //~ ERROR E0495
   |                                           ^
   |
   = note: first, the lifetime cannot outlive lifetime RePlaceholder(Placeholder { universe: U1, name: BrAnon(1) })...
   = note: ...but the lifetime must also be valid for lifetime RePlaceholder(Placeholder { universe: U1, name: BrAnon(1) })...
   = note: ...so that the expression is assignable:
           expected for<'r, 's, 't0, 't1> fn(&'r mut &'s isize, &'t0 mut &'t1 isize)
              found for<'r, 's> fn(&'r mut &isize, &'s mut &isize)

error: aborting due to 3 previous errors

Some errors occurred: E0495, E0623.
For more information about an error, try `rustc --explain E0495`.
