error: character literal may only contain one codepoint
  --> $DIR/lex-bad-char-literals-3.rs:3:5
   |
LL |     '●●' //~ ERROR: character literal may only contain one codepoint
   |     ^^^^
help: if you meant to write a `str` literal, use double quotes
   |
LL |     "●●" //~ ERROR: character literal may only contain one codepoint
   |     ^^^^

error: character literal may only contain one codepoint
  --> $DIR/lex-bad-char-literals-3.rs:7:20
   |
LL |     let ch: &str = '●●'; //~ ERROR: character literal may only contain one codepoint
   |                    ^^^^
help: if you meant to write a `str` literal, use double quotes
   |
LL |     let ch: &str = "●●"; //~ ERROR: character literal may only contain one codepoint
   |                    ^^^^

error[E0308]: mismatched types
  --> $DIR/lex-bad-char-literals-3.rs:7:20
   |
LL |     let ch: &str = '●●'; //~ ERROR: character literal may only contain one codepoint
   |                    ^^^^ expected &str, found char
   |
   = note: expected type `&str`
              found type `char`

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0308`.
