error: character literal may only contain one codepoint
  --> $DIR/lex-bad-char-literals-6.rs:2:19
   |
LL |     let x: &str = 'ab';  //~ ERROR: character literal may only contain one codepoint
   |                   ^^^^

error: character literal may only contain one codepoint
  --> $DIR/lex-bad-char-literals-6.rs:3:19
   |
LL |     let y: char = 'cd';  //~ ERROR: character literal may only contain one codepoint
   |                   ^^^^

error: character literal may only contain one codepoint
  --> $DIR/lex-bad-char-literals-6.rs:4:13
   |
LL |     let z = 'ef';  //~ ERROR: character literal may only contain one codepoint
   |             ^^^^

error[E0308]: mismatched types
  --> $DIR/lex-bad-char-literals-6.rs:2:19
   |
LL |     let x: &str = 'ab';  //~ ERROR: character literal may only contain one codepoint
   |                   ^^^^ expected &str, found char
   |
   = note: expected type `&str`
              found type `char`

error[E0277]: can't compare `&str` with `char`
  --> $DIR/lex-bad-char-literals-6.rs:6:10
   |
LL |     if x == y {}  // no error here
   |          ^^ no implementation for `&str == char`
   |
   = help: the trait `std::cmp::PartialEq<char>` is not implemented for `&str`

error[E0308]: mismatched types
  --> $DIR/lex-bad-char-literals-6.rs:10:20
   |
LL |     let a: usize = "";  // type error here to confirm we got past the parser
   |                    ^^ expected usize, found reference
   |
   = note: expected type `usize`
              found type `&'static str`

error[E0277]: can't compare `&str` with `char`
  --> $DIR/lex-bad-char-literals-6.rs:8:10
   |
LL |     if x == z {}  // no error here
   |          ^^ no implementation for `&str == char`
   |
   = help: the trait `std::cmp::PartialEq<char>` is not implemented for `&str`

error: aborting due to 7 previous errors

Some errors occurred: E0277, E0308.
For more information about an error, try `rustc --explain E0277`.
