error: multiple unused formatting arguments
  --> $DIR/format-unused-lables.rs:12:22
   |
12 |     println!("Test", 123, 456, 789);
   |                      ^^^  ^^^  ^^^

error: multiple unused formatting arguments
  --> $DIR/format-unused-lables.rs:16:9
   |
16 |         123,  //~ ERROR multiple unused formatting arguments
   |         ^^^
17 |         456,
   |         ^^^
18 |         789
   |         ^^^

error: named argument never used
  --> $DIR/format-unused-lables.rs:21:35
   |
21 |     println!("Some stuff", UNUSED="args"); //~ ERROR named argument never used
   |                                   ^^^^^^

error: multiple unused formatting arguments
  --> $DIR/format-unused-lables.rs:24:9
   |
24 |         "woo!",  //~ ERROR multiple unused formatting arguments
   |         ^^^^^^
25 |             STUFF=
26 |        "things"
   |        ^^^^^^^^
27 |              , UNUSED="args");
   |                       ^^^^^^
   |
   = help: `$STUFF` should be written as `{STUFF}`
   = note: shell formatting not supported; see the documentation for `std::fmt`

error: aborting due to 4 previous errors

