error: `?` macro repetition does not allow a separator
  --> $DIR/macro-at-most-once-rep-ambig.rs:29:10
   |
LL |     ($(a),?) => {} //~ ERROR `?` macro repetition does not allow a separator
   |          ^

error: no rules expected the token `?`
  --> $DIR/macro-at-most-once-rep-ambig.rs:43:11
   |
LL |     foo!(a?a?a); //~ ERROR no rules expected the token `?`
   |           ^

error: no rules expected the token `?`
  --> $DIR/macro-at-most-once-rep-ambig.rs:44:11
   |
LL |     foo!(a?a); //~ ERROR no rules expected the token `?`
   |           ^

error: no rules expected the token `?`
  --> $DIR/macro-at-most-once-rep-ambig.rs:45:11
   |
LL |     foo!(a?); //~ ERROR no rules expected the token `?`
   |           ^

error: unexpected end of macro invocation
  --> $DIR/macro-at-most-once-rep-ambig.rs:46:5
   |
LL |     barplus!(); //~ ERROR unexpected end of macro invocation
   |     ^^^^^^^^^^^

error: unexpected end of macro invocation
  --> $DIR/macro-at-most-once-rep-ambig.rs:47:5
   |
LL |     barstar!(); //~ ERROR unexpected end of macro invocation
   |     ^^^^^^^^^^^

error: no rules expected the token `?`
  --> $DIR/macro-at-most-once-rep-ambig.rs:48:15
   |
LL |     barplus!(a?); //~ ERROR no rules expected the token `?`
   |               ^

error: unexpected end of macro invocation
  --> $DIR/macro-at-most-once-rep-ambig.rs:49:14
   |
LL |     barplus!(a); //~ ERROR unexpected end of macro invocation
   |              ^

error: no rules expected the token `?`
  --> $DIR/macro-at-most-once-rep-ambig.rs:50:15
   |
LL |     barstar!(a?); //~ ERROR no rules expected the token `?`
   |               ^

error: unexpected end of macro invocation
  --> $DIR/macro-at-most-once-rep-ambig.rs:51:14
   |
LL |     barstar!(a); //~ ERROR unexpected end of macro invocation
   |              ^

error: aborting due to 10 previous errors

