error[E0308]: mismatched types
  --> $DIR/hrtb-just-for-static.rs:24:5
   |
LL |     want_hrtb::<StaticInt>() //~ ERROR
   |     ^^^^^^^^^^^^^^^^^^^^^^ lifetime mismatch
   |
   = note: expected type `Foo<&'a isize>`
              found type `Foo<&'static isize>`
   = note: lifetime RePlaceholder(Placeholder { universe: U1, name: BrNamed(crate0:DefIndex(1:11), 'a) })...
   = note: ...does not necessarily outlive the static lifetime

error: implementation of `Foo` is not general enough
  --> $DIR/hrtb-just-for-static.rs:30:5
   |
LL |     want_hrtb::<&'a u32>() //~ ERROR
   |     ^^^^^^^^^^^^^^^^^^^^
   |
   = note: Due to a where-clause on `want_hrtb`,
   = note: `&'a u32` must implement `Foo<&'0 isize>` for any lifetime `'0`
   = note: but `&'1 u32` only implements `Foo<&'1 isize>` for some lifetime `'1`

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0308`.
